---
title: "septoria_in_a_bag_fungal_statistical_analyses"
output: html_document
date: "2024-02-06"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Loading Packages

```{r}
library(vegan)
library(ggplot2)
library(dplyr)
library(ape)
library(tidyr)
library(DESeq2)
library(ggpubr)
library(hillR)
library(rstatix)
library(car)
library(pime)
library(phyloseq)
library(stringr)
library(FSA)
library(plotly)
```

# Creating a rarefy table function

The below function takes a rarfied table as input and removes samples that do not have the minimum sampling depth and any ASV that has a sum across samples of zeros following rarefaction. The function takes as input the following items: 

1. **rarefied_table**: a rarefied ASV table with unrarefied samples and zero sum ASVs still present
2. **sample**: sequencing depth by which all samples were rarefied.

```{r}
# Removing unrarefied samples (those less than the minimum sequencing depth). Removing ASVs present in the ASV table that were only present in samples removed following rarefaction. These ASVs will have column sums of 0. 
rarefy_formatting<-function(rarefied_table,sample){
  ds.rarefied<-as.data.frame(subset(rarefied_table, rowSums(rarefied_table)>=sample))
  ds.rare.asv<-ds.rarefied[,colSums(ds.rarefied)>0]
  return(ds.rare.asv)
}
```

# Creating a PCoA Function

The below function wraps the vegdist function from vegan and the pcoa function from ape and generates an ordination using ggplot. The results of the PCoA and the ggplot are stored in a two object list. The first is the PCoA output and the second is the ggplot that can be further customized using ggplot arguments. The function requires the following input:

1. **table**: a rarefied ASV table
2. **method**: a method for computing distances (see: ??vegan::vegdist for options)
3. **point_ellipse_color**: The variable by which points and ellipses should be colored
4. **shape**: Shaping variable
5. **polygon**: Logical value indiciating whether filled ellipses should be plotted to reprsented. This is useful when working with data with multiple factors. **Must be specified with ellipse_fill has a value.** 
6. **ellipse_fill**: Variable by which filled ellipses should be colored. 
7. **point_size**: Size of points for ordination. Default is 3.5

```{r}
pcoa_imager<-function(table,method,point_ellipse_color,shape=NULL,polygon=NULL,ellipse_fill=NULL,point_size=3.5){
  column_grabber<-`[[` # Generating a subsetting function
  point_ellipse_color<-column_grabber(table,point_ellipse_color) # will subset the column that contains the provided variable by which points and ellipses should be colored
  ellipse_fill<-column_grabber(table,ellipse_fill) # will subset the column that contains the variabe by which ellipses will be filled
  shape<-column_grabber(table,shape) # will subset the column that contains the point shaping variabe 
  relabund.asv<-decostand(Filter(x=table,f=is.numeric), method="total") # using the decostand function from vegan to convert count data to relative abundances
  dist.asv<-vegdist(relabund.asv, method=method) # calculating distance matrix using the vegdist function from vegan. Method is a user specified variable to indicate which method for computing distance to be used. 
  pcoa.asv<-pcoa(dist.asv) # using the pcoa function from ape to perform a PCoA with the distance matrix stored in dist.asv. 
  pcoavec.asv<-as.data.frame(pcoa.asv$vectors) # creating a dataframe that only includes the vectors from the PCoA.
  pcoasitescores.asv<-data.frame(PC1=pcoavec.asv$Axis.1, PC2=pcoavec.asv$Axis.2) # creating a datatable that is only the first and second PCs. 
  
  # creating a if/else statement based on presence of ellipse_fill. 
      if (is.null(ellipse_fill)){
      pcoagraph.asv<-data.frame(pcoasitescores.asv,PC1=pcoasitescores.asv$PC1, PC2=pcoasitescores.asv$PC2, color=point_ellipse_color, shape_var=shape)
    } else {
  pcoagraph.asv<-data.frame(pcoasitescores.asv,PC1=pcoasitescores.asv$PC1, PC2=pcoasitescores.asv$PC2, color=point_ellipse_color, ellipse_fill=ellipse_fill, shape_var=shape)}
  
  # Generates standard deviation ellipses based on color variable
  pcoaellipse<-ordiellipse(pcoasitescores.asv,pcoagraph.asv$color, display="sites", kind="sd", draw="none")
  ell <- data.frame()
for(g in levels(as.factor(pcoagraph.asv$color))){
ell <- rbind(ell, cbind(as.data.frame(with(pcoagraph.asv[pcoagraph.asv$color==g,],                                                vegan:::veganCovEllipse(pcoaellipse[[g]]$cov,pcoaellipse[[g]]$center,pcoaellipse[[g]]$scale))) ,color=g))}
  
  # Generates secondary ellipse for second color
pcoaellipse.2<-ordiellipse(pcoasitescores.asv,pcoagraph.asv$ellipse_fill, display="sites", kind="sd", draw="none")
  ell_2 <- data.frame()
for(g in levels(as.factor(pcoagraph.asv$ellipse_fill))){
ell_2 <- rbind(ell_2, cbind(as.data.frame(with(pcoagraph.asv[pcoagraph.asv$elllipse_fill==g,],                                                vegan:::veganCovEllipse(pcoaellipse.2[[g]]$cov,pcoaellipse.2[[g]]$center,pcoaellipse.2[[g]]$scale))) ,ellipse_fill=g))}  
  
# Plotting first two PCs of PCoA in ggplot  
  
  # If an ellipse_fill variable is present polygon must be set to true
  
  if (polygon==TRUE){
  pcoa_plot<-ggplot(pcoagraph.asv, aes(PC1,PC2, colour=color))+
  # If statement that indicates what to do in the absence of a shape variable
    if (is.null(shape)){
      geom_point(size=point_size)+
      geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
  geom_polygon(data=ell_2, aes(x=PC1, y=PC2, fill=color.two),linewidth=0.5, linetype=2,inherit.aes = FALSE,alpha=0.1)+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", linewidth =1, fill=NA),
    panel.background = element_rect(fill="white"))+
    xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))
   
       } else {
         
  geom_point(aes(shape=shape_var),size=point_size)+
  geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
  geom_polygon(data=ell_2, aes(x=PC1, y=PC2, fill=color.two),linewidth=0.5, linetype=2,inherit.aes = FALSE,alpha=0.1)+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", linewidth =1, fill=NA),
    panel.background = element_rect(fill="white"))+
    xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))}
         
    } else {
      
      if (is.null(shape)){
      geom_point(size=point_size)+
      geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", linewidth =1, fill=NA),
    panel.background = element_rect(fill="white"))+
    xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))
      
        } else {
  
  geom_point(aes(shape=shape_var), size=point_size)+
  geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
 theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA),
    panel.background = element_rect(fill="white"))+
      xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))}
    }
  pcoa_func_out<-list(pcoa.asv,pcoa_plot)
  return(list(pcoa.asv,pcoa_plot))
}

```

# Relative Abundance Plot Function

A function to generates relative abundance bar charts that requires the following input:

1. **metadata**: The object containing metadata info to be used for grouping data.
2. **taxonomy.table**: Taxonomy table generated by the dada2 pipeline
3. **asv.table**: Rarefied ASV table
4. **grouping.var**: Grouping variable present in the metadata object 
5. **taxon_level**: Taxonomy level to be used for plotting
6. **cut_off**: Relative abundance cut-off to be shown in the chart

```{r}
relative_abundance_plots<-function(metadata,taxonomy.table,asv.table,grouping.var,taxon_level,cut_off,kingdom){
  if(kingdom=="fungi"){
    taxonomy.table[["Kingdom"]]<-sub("k__","",taxonomy.table[["Kingdom"]])
    taxonomy.table[["Phylum"]]<-sub("p__","",taxonomy.table[["Phylum"]])
    taxonomy.table[["Class"]]<-sub("c__","",taxonomy.table[["Class"]])
    taxonomy.table[["Order"]]<-sub("o__","",taxonomy.table[["Order"]])
    taxonomy.table[["Family"]]<-sub("f__","",taxonomy.table[["Family"]])
    taxonomy.table[["Genus"]]<-sub("g__","",taxonomy.table[["Genus"]])
    taxonomy.table[["Species"]]<-sub("s__","",taxonomy.table[["Species"]])
  }
  metadata.asv.tab<-merge(metadata,asv.table,by.x=0,by.y=0) # Merges asv table and metadata file by row names
  metadata.asv.tab_group_sums<-metadata.asv.tab[,c(which(colnames(metadata.asv.tab)==paste0("",deparse(substitute(grouping.var)),"")),which(sapply(metadata.asv.tab,is.numeric)))]%>% # Takes the sum of each ASV by the grouping variable
    as.data.frame()%>%
    group_by({{grouping.var}})%>%
    summarize_all(sum)%>%
    as.data.frame()
  row.names(metadata.asv.tab_group_sums)<-metadata.asv.tab_group_sums[[deparse(substitute(grouping.var))]] # Converts grouping variable to rownames
  metadata.asv.tab_group_sums[[deparse(substitute(grouping.var))]]<-NULL # Removes duplicate column 
  t.metadata.asv.tab_group_sums<-as.data.frame(t(metadata.asv.tab_group_sums)) # Transposes group sum data frame
 t.metadata.asv.tab_group_sums[]<-as.data.frame(sapply(t.metadata.asv.tab_group_sums,as.numeric)) # Converts data to numeric
 metadata.asv.tab.taxonomy_group_sums<-merge(taxonomy.table,t.metadata.asv.tab_group_sums,by=0) # Merges taxonomy information into group summed ASV table
 metadata.asv.tab.taxonomy_taxon_sums<-metadata.asv.tab.taxonomy_group_sums[,c(which(colnames(metadata.asv.tab.taxonomy_group_sums)==paste0("",deparse(substitute(taxon_level)),"")),which(sapply(metadata.asv.tab.taxonomy_group_sums,is.numeric)))]%>% # Takes the sum of ASVs by their taxonomic assignment
 as.data.frame()%>%
group_by({{taxon_level}})%>%
summarize_all(sum)%>%
as.data.frame()

 metadata.asv.tab.taxonomy_taxon_sums[[deparse(substitute(taxon_level))]][which(is.na(metadata.asv.tab.taxonomy_taxon_sums[[deparse(substitute(taxon_level))]]))]<-"Unassigned" # Renames NA assignments as unassigned 
  rownames(metadata.asv.tab.taxonomy_taxon_sums)<-metadata.asv.tab.taxonomy_taxon_sums[[deparse(substitute(taxon_level))]] # Updates row names to taxonomy names
  metadata.asv.tab.taxonomy_taxon_sums[[deparse(substitute(taxon_level))]]<-NULL # Removes duplicate column
  t.metadata.asv.tab.taxonomy_taxon_sums<-as.data.frame(t(metadata.asv.tab.taxonomy_taxon_sums)) # Transposes data frame 
  t.metadata.asv.tab.taxonomy_taxon_cutoff<-t.metadata.asv.tab.taxonomy_taxon_sums[,which((colSums(t.metadata.asv.tab.taxonomy_taxon_sums[,grep("Unassigned",colnames(t.metadata.asv.tab.taxonomy_taxon_sums),invert=TRUE)])/sum(t.metadata.asv.tab.taxonomy_taxon_sums))>substitute(cut_off))] # Creates an object that has only those ASVs that have a relative abundance greater than the specified cut-off
    t.metadata.asv.tab.taxonomy_taxon_other<-t.metadata.asv.tab.taxonomy_taxon_sums[,which((colSums(t.metadata.asv.tab.taxonomy_taxon_sums[,grep("Unassigned",colnames(t.metadata.asv.tab.taxonomy_taxon_sums),invert=TRUE)])/sum(t.metadata.asv.tab.taxonomy_taxon_sums))<substitute(cut_off))] # Creates an object that contains only the taxa who fell below the specified relative abundance cut-off
  taxon_unassigned<-data.frame(Unassigned=t.metadata.asv.tab.taxonomy_taxon_sums[,grep("Unassigned",colnames(t.metadata.asv.tab.taxonomy_taxon_sums),invert=FALSE)]) # Creates an object to store the unassigned taxa
  other<-data.frame(Other=rowSums( t.metadata.asv.tab.taxonomy_taxon_other)) # Takes the sum of the taxa that fell below the cut-off
  t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund<-decostand(cbind(t.metadata.asv.tab.taxonomy_taxon_cutoff,other,taxon_unassigned),method="total") # Calculates relative abundance 
  t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund_long<-t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund%>%
   tibble::rownames_to_column(var="Treatment")%>%
   pivot_longer(cols=!Treatment,names_to = "taxon_level_plot", values_to = "Relative.Abundance")%>%
    as.data.frame() # Converts data frame to long
  
t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund_long[["taxon_level_plot"]]<-factor(unique(t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund_long[["taxon_level_plot"]]),levels=unique(t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund_long[["taxon_level_plot"]])) # Converts 

colour_count<-length(unique(t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund_long[["taxon_level_plot"]]))
getPalette <- colorRampPalette(RColorBrewer::brewer.pal(9,"Set1"))
  
return(ggplot(t.metadata.asv.tab.taxonomy_taxon_cutoff_relabund_long,aes(x=Treatment,y=Relative.Abundance))+
geom_col(aes(x=Treatment,y=Relative.Abundance,fill=taxon_level_plot),linewidth=0.75,inherit.aes = FALSE,color="black")+
theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),legend.position = "bottom")+
  scale_fill_manual(values = c(getPalette(colour_count-1),"lightgrey"))+
guides(fill=guide_legend(title = deparse(substitute(taxon_level)))))
}

```

# Loading ASV Table and Data Formatting

Loading in the fungal ASV table that contains the taxonomy information. Then splitting the taxonomy information from the ASV table. 

```{r}
# Loading in ASV table with taxonomy information. ASV table has a column called Row.names which is the ASV sequences. Negative controls are present in this table. 
t.seqtab.tax_its_original<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/soil_asv_table_septoria_in_a_bag_its.txt",header=TRUE,sep='\t')

# Checking the row names. Should be ASV IDs
row.names(t.seqtab.tax_its_original)

# Checking the column names. Should be sample names
colnames(t.seqtab.tax_its_original)

# Removing generic illumina info from sample names
colnames(t.seqtab.tax_its_original)<-sub("_F_filt.fastq","",colnames(t.seqtab.tax_its_original))

# Filtering out ASVs not assigned to a fungal phyla
t.seqtab.tax_its.na_omit<-t.seqtab.tax_its_original[-(which(is.na(t.seqtab.tax_its_original$Phylum))),]
t.seqtab.tax_its<-t.seqtab.tax_its.na_omit[grep("k__Fungi",t.seqtab.tax_its.na_omit$Kingdom),]

# Number of ASVs in the study following removal of unassigned taxa and non-fungal taxa.
nrow(t.seqtab.tax_its)

# Removing Negatives from ASV table
t.seqtab.tax_its<-t.seqtab.tax_its[,grep("NTC",colnames(t.seqtab.tax_its),invert=TRUE)]

# Subsetting data frame so that only ASV count info is present in ASV table
t.seqtab_its<-t.seqtab.tax_its[,2:(length(t.seqtab.tax_its)-7)]
sum(t.seqtab_its)

# Subsetting data frame so that only taxonomy is present 
tax.tab_its<-t.seqtab.tax_its[,(length(t.seqtab.tax_its)-6):(length(t.seqtab.tax_its))]

#write.table(tax.tab_its,"~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/tax.tab_its.txt",row.names=TRUE,col.names = TRUE,sep='\t')

# Transposing ASV table back to original format so that it can be used with vegan. 
seqtab_its<-as.data.frame(t(t.seqtab_its))

# Fixing the sample name
row.names(seqtab_its)[59]<-sub("NoSept_NA_P_R2_MoP_D25","NoSept_NA_NP_R2_MoP_D25",row.names(seqtab_its)[59])

# List of ASVs identified as Sphaerulina musiva by BLAST
sphaerulina_asvs<-c("asv_its 360","asv_its 454","asv_its 1237","asv_its 1238","asv_its 1239","asv_its 1240","asv_its 1249","asv_its 1288","asv_its 1352","asv_its 2539","asv_its 3962")
```

# Metadata Object

Generating a metadata object that includes the following information:

1. **Inoculation Type**
  A. SARM07: *Sphaerulina musiva* low virulence
  B. Mn5: *Sphaerulina musiva* high virulence
  C. NoSept: Non-inoculate soils

2. **Autoclave Type**
  A. A: Field soils collected from Oregon that were autoclaved prior to the start of the study and amended into the soil. 
  B. NA: Field soils collected from Oregon that were amended into study soil that were not autoclaved.
  
3. **Plant Presence**
  A. P: *Populus* was present in the pot.
  B. NP: *Populus* was not present in the pot and it was soil only. 
  
4. **Location of Soil Collection**
  A. EoP: Edge of pot 
  B. MoP: Middle of pot
  C. NR: Near root
  
5. **Date of Collection**
  A. D10/D25: Samples with this information present in the beginning of the name were the initial samples collected prior to fungal addition. 
  B. o9: Sequencing was completed for these samples October 9th (likely Day 10 samples)
  C. o15: Sequencing was completed for these samples October 15th (likely Day 25 samples)
  
```{r}
# Creating an empty matrix to ad lib a metadata file
s.bag.meta<-matrix(nrow=286,ncol=1)
s.bag.meta<-as.data.frame(s.bag.meta)
row.names(s.bag.meta)<-row.names(seqtab_its)
s.bag.meta$sample_names<-as.data.frame(row.names(seqtab_its))

# Splitting out information about septoria inoculation from sample names
septoria<-c(sapply(strsplit(row.names(seqtab_its[1:192,]),"_"),'[',1),sapply(strsplit(row.names(seqtab_its[193:nrow(seqtab_its),]),"_"),'[',2))

# Splitting out information about soil autoclaving from sample names
autoclave<-c(sapply(strsplit(row.names(seqtab_its[1:192,]),"_"),'[',2),sapply(strsplit(row.names(seqtab_its[193:nrow(seqtab_its),]),"_"),'[',3))

# Splitting out information about presence of a plant from sample names
plant<-c(sapply(strsplit(row.names(seqtab_its[1:192,]),"_"),'[',3),sapply(strsplit(row.names(seqtab_its[193:nrow(seqtab_its),]),"_"),'[',4))

# Splitting out information about reps from sample names
rep<-c(sapply(strsplit(row.names(seqtab_its[1:192,]),"_"),'[',4),sapply(strsplit(row.names(seqtab_its[193:nrow(seqtab_its),]),"_"),'[',5))

# Splitting out information regarding sample location collection information
location<-c(sapply(strsplit(row.names(seqtab_its[1:192,]),"_"),'[',5),rep("initial",times=94))

# Splitting out information for sampling date
date<-c(sapply(strsplit(row.names(seqtab_its[1:192,]),"_"),'[',6),sapply(strsplit(row.names(seqtab_its[193:nrow(seqtab_its),]),"_"),'[',6))

# Adding the metadata to the metadata sheet
s.bag.meta$septoria<-septoria
s.bag.meta$autoclave<-autoclave
s.bag.meta$plant<-plant
s.bag.meta$location<-location
s.bag.meta$rep<-rep
s.bag.meta$date<-date
s.bag.meta<-s.bag.meta[,-c(1,2)]

# Fixing one of the septoria labels that were spelt a little different from the others.
s.bag.meta$septoria<-sub("noSept","NoSept",s.bag.meta$septoria)
sample.metadata<-s.bag.meta

# Fixing one of the sample dates that had B instead of D
sample.metadata$date<-sub("B","D25",sample.metadata$date)

#sample_metadata$sept_general<-ifelse(grepl("Mn5|SARM07",sample_metadata$septoria)==TRUE,"septoria","control")

# Creating a new category called plant_date that indicates plant treatment and date of collection.
sample.metadata$plant_date<-paste0(sample.metadata$plant,".",sample.metadata$date)

# Creating a new category called plant_date_septoria that indicates plant treatment, date of collection, sphaerulina inoculant.
sample.metadata$date_septoria<-paste0(sample.metadata$date,"_",sample.metadata$septoria)

# Crating a new category called sept_autoclave that indicates sphaerulina inoculant and autoclave treatment. 
sample.metadata$sept_autoclave<-paste0(sample.metadata$septoria,"_",sample.metadata$autoclave)

# Creating a new category called sept_plant that indicates sphaerulina incoulant and plant treatment
sample.metadata$sept_plant<-paste0(sample.metadata$plant,"_",sample.metadata$septoria)

# Determining number of reps per treatment type
as.data.frame(sample.metadata) %>%
  group_by(date, septoria, autoclave,plant,location) %>%
  dplyr::count() %>%
  print(n=61)

# Writing metadata to file
#write.table(sample.metadata,"~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sample_metadata_from_sample_names.txt",row.names=TRUE,col.names=TRUE,sep='\t')

as.data.frame(sample.metadata_nr) %>%
  group_by(date, septoria, autoclave,plant,location) %>%
  dplyr::count() %>%
  print(n=61)
```

# Generating rarefaction curves for all samples

```{r}
# Generating rarefaction curves for all samples
rarefy.plot_its<-rarecurve(seqtab_its,tidy = TRUE)

# Plotting rarefaction curves in ggplot
rare.plot_its<-ggplot()+
  geom_line(data=rarefy.plot_its,aes(x=Sample,y=Species,color=Site))+
  geom_vline(data=NULL,xintercept=c(2000,5000,10000,15000,20000,30000,40000))+
  theme(legend.position="none")+
  geom_text(data=NULL,aes(x=1000,y=1500,label="2000",angle=90))+
  geom_text(data=NULL,aes(x=4000,y=1500,label="5000",angle=90))+
  geom_text(data=NULL,aes(x=9000,y=1500,label="10000",angle=90))+
  geom_text(data=NULL,aes(x=14000,y=1500,label="15000",angle=90))+
  geom_text(data=NULL,aes(x=19000,y=1500,label="20000",angle=90))+
  geom_text(data=NULL,aes(x=29000,y=1500,label="30000",angle=90))+
  geom_text(data=NULL,aes(x=39000,y=1500,label="40000",angle=90))

rare.plot_its

# Writing rarefy data to file
# write.table(rarefy.plot_its,"~/OneDrive - Oak Ridge National Laboratory/sphaerulina_soil_inoculation_project/fungal_its2_data_processing/rarefy.plot_its.csv",row.names=TRUE,col.names=TRUE,sep='/,')

# Subsetting overall plot to look at autoclave only to help determine what samples need to be redone. 
rarefy.plot_its_autoclave<-rarefy.plot_its[grep("_A_",rarefy.plot_its$Site),]

# Plotting rarefaction curves in ggplot
rare.plot_its_autoclave<-ggplot()+
  geom_line(data=rarefy.plot_its_autoclave,aes(x=Sample,y=Species,color=Site))+
  geom_vline(data=NULL,xintercept=c(2000,5000,10000,15000,20000,30000,40000))+
  theme(legend.position="none")+
  geom_text(data=NULL,aes(x=1000,y=1500,label="2000",angle=90))+
  geom_text(data=NULL,aes(x=4000,y=1500,label="5000",angle=90))+
  geom_text(data=NULL,aes(x=9000,y=1500,label="10000",angle=90))+
  geom_text(data=NULL,aes(x=14000,y=1500,label="15000",angle=90))+
  geom_text(data=NULL,aes(x=19000,y=1500,label="20000",angle=90))+
  geom_text(data=NULL,aes(x=29000,y=1500,label="30000",angle=90))+
  geom_text(data=NULL,aes(x=39000,y=1500,label="40000",angle=90))

rare.plot_its_autoclave

# Subsetting overall plot to look at non-autoclave only to help determine what samples need to be redone. 
rarefy.plot_its_nonautoclave<-rarefy.plot_its[grep("_NA_",rarefy.plot_its$Site),]

# Plotting rarefaction curves in ggplot
rare.plot_its_nonautoclave<-ggplot()+
  geom_line(data=rarefy.plot_its_nonautoclave,aes(x=Sample,y=Species,color=Site))+
  geom_vline(data=NULL,xintercept=c(2000,5000,10000,15000,20000,30000,40000))+
  theme(legend.position="none")+
  geom_text(data=NULL,aes(x=1000,y=1500,label="2000",angle=90))+
  geom_text(data=NULL,aes(x=4000,y=1500,label="5000",angle=90))+
  geom_text(data=NULL,aes(x=9000,y=1500,label="10000",angle=90))+
  geom_text(data=NULL,aes(x=14000,y=1500,label="15000",angle=90))+
  geom_text(data=NULL,aes(x=19000,y=1500,label="20000",angle=90))+
  geom_text(data=NULL,aes(x=29000,y=1500,label="30000",angle=90))+
  geom_text(data=NULL,aes(x=39000,y=1500,label="40000",angle=90))

rare.plot_its_nonautoclave


# Rarefying the ASV table to 20000 sequences
rare.asv.tab_its<-as.data.frame(rrarefy(seqtab_its, sample=20000))

# Removing samples that do not meet rarefaction cut-off and ASVs with zero reads following rarefaction.
rare.asv.tab.fmt_its<-rarefy_formatting(rare.asv.tab_its,20000)

# Exporting rarefied ASV table to have a consitent file
#write.table(rare.asv.tab.fmt_its,"~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sphaerulina_soil_inoculations/its_v4_statistical_analyses/rdata_its/rare.asv.tab.fmt_its.txt",col.names = TRUE,row.names = TRUE,sep='\t')

```

```{r}
# Subsetting metadata to include only day 10 and day 25 samples 
sample.metadata_d10d25<-sample.metadata[grep("D10|D25",sample.metadata$date),]

# Subsetting metadata to include only near root/middle of pot samples. 
sample.metadata_nr<-sample.metadata_d10d25[grep("NR|MoP",sample.metadata_d10d25$location),]

# Subsetting to include only day 10 and day 25 samples
seqtab_its_d10d25<-seqtab_its[grep("_D10|_D25",row.names(seqtab_its)),]

# Subsetting to include only near root samples. 
seqtab_its_d10d25_nrmop<-seqtab_its_d10d25[grep("_NR_|_MoP_|Mn5_A_NP_R2_EoP_D25|Mn5_A_NP_R4_EoP_D25|NoSept_A_NP_R1_EoP_D25|NoSept_A_NP_R2_EoP_D25|SARM07_A_NP_R1_EoP_D25|Mn5_NA_P_R1_EoP_D10|Mn5_NA_P_R4_EoP_D10|Mn5_NA_P_R3_EoP_D25|Mn5_NA_NP_R2_EoP_D10",row.names(seqtab_its_d10d25)),]
nrow(seqtab_its_d10d25_nrmop)

# Checking sample counts. All should have 4 except for D25 A/NP plants which will have 5 to 6 to try and account for low sequencing depth in MoP samples
as.data.frame(merge(seqtab_its_d10d25_nrmop,sample.metadata,by.x=0,by.y=0)) %>%
  group_by(date, septoria, autoclave,plant) %>%
  dplyr::count() %>%
  print(n=61)

# Plotting rarefaction curves in ggplot
rareplot_its_d10d25_nrmop<-ggplot()+
  geom_line(data=rarefy.plot_its_autoclave,aes(x=Sample,y=Species,color=Site))+
  geom_vline(data=NULL,xintercept=c(2000,5000,10000,15000,20000,30000,40000))+
  theme(legend.position="none")+
  geom_text(data=NULL,aes(x=1000,y=1500,label="2000",angle=90))+
  geom_text(data=NULL,aes(x=4000,y=1500,label="5000",angle=90))+
  geom_text(data=NULL,aes(x=9000,y=1500,label="10000",angle=90))+
  geom_text(data=NULL,aes(x=14000,y=1500,label="15000",angle=90))+
  geom_text(data=NULL,aes(x=19000,y=1500,label="20000",angle=90))+
  geom_text(data=NULL,aes(x=29000,y=1500,label="30000",angle=90))+
  geom_text(data=NULL,aes(x=39000,y=1500,label="40000",angle=90))

rareplot_its_d10d25_nrmop

# Rarefying the ASV table to 10000 sequences
rare.asv.tab_its_d10d25_nrmop<-as.data.frame(rrarefy(seqtab_its_d10d25_nrmop, sample=10000))

# Removing samples that do not meet rarefaction cut-off and ASVs with zero reads following rarefaction.
rare.asv.tab.fmt_its_d10d25_nrmop<-rarefy_formatting(rare.asv.tab_its_d10d25_nrmop,10000)

# Exporting rarefied ASV table to have a consitent file
#write.table(rare.asv.tab.fmt_its_d10d25_nrmop,"~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10_d25_nrmop.txt",col.names = TRUE,row.names = TRUE,sep='\t')

```

# Starting statistical analyses of d10/d25 data 

```{r}
# Loading in rarefied ASV table
rare.asv.tab.fmt_its_d10d25_nrmop<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10_d25_nrmop.txt",header=TRUE,sep='\t')

# Loading in sample metadata that we made from sample names
sample.metadata<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sample_metadata_from_sample_names.txt",header=TRUE,sep='\t',na.strings = "-")

# Adding metadata to ASV table.
metadata.rare.asv.tab.fmt_its_d10d25_nrmop<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop,by.x=0,by.y=0)

# Checking number of samples retained after rarefaction. 
nrow(metadata.rare.asv.tab.fmt_its_d10d25_nrmop)

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_d10d25_nrmop %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Won't have enough power to look at plant/isolate/autoclave interaction but will have enough to look at autoclave alone. 

# General Beta-Diversity
pcoa_its_autoclave<-pcoa_imager(metadata.rare.asv.tab.fmt_its_d10d25_nrmop,method = "bray",color="autoclave",shape="septoria", polygon=TRUE,color_two="autoclave")

pcoa_its_autoclave[[1]]
pcoa_its_autoclave_plot<-pcoa_its_autoclave[[2]]+
  guides(fill=guide_legend("Autoclave"),color=guide_legend("Autoclave"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("Autoclaved","Non-Autoclaved"))+
  scale_color_discrete(labels=c("Autoclaved","Non-Autoclaved"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_autoclave_plot

adonis2(Filter(x=metadata.rare.asv.tab.fmt_its_d10d25_nrmop,f=is.numeric)~metadata.rare.asv.tab.fmt_its_d10d25_nrmop$autoclave)

# General Alpha-Diversity
alpha_diversity_d10_d25_nr_mop<-data.frame(select_if(metadata.rare.asv.tab.fmt_its_d10d25_nrmop,is.character),q0=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_d10d25_nrmop,is.numeric),q=0),q1=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_d10d25_nrmop,is.numeric),q=1),q2=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_d10d25_nrmop,is.numeric),q=2))


# T.test/Wilcox Tests for q0
t.test(formula=alpha_diversity_d10_d25_nr_mop$q0~alpha_diversity_d10_d25_nr_mop$autoclave)
shapiro.test(alpha_diversity_d10_d25_nr_mop$q0)
ggqqplot(alpha_diversity_d10_d25_nr_mop,x="q0")

# Non-parametric alternative
wilcox.test(alpha_diversity_d10_d25_nr_mop$q0~alpha_diversity_d10_d25_nr_mop$autoclave)
wilcox_effsize(alpha_diversity_d10_d25_nr_mop,q0~autoclave)

# T.test/Wilcox Tests for q1
t.test(formula=alpha_diversity_d10_d25_nr_mop$q1~alpha_diversity_d10_d25_nr_mop$autoclave)
shapiro.test(alpha_diversity_d10_d25_nr_mop$q1)
ggqqplot(alpha_diversity_d10_d25_nr_mop,x="q1")

# Non-parametric alternative
wilcox.test(alpha_diversity_d10_d25_nr_mop$q1~alpha_diversity_d10_d25_nr_mop$autoclave)
wilcox_effsize(alpha_diversity_d10_d25_nr_mop,q1~autoclave)

# T.test/Wilcox Tests for q2
t.test(formula=alpha_diversity_d10_d25_nr_mop$q2~alpha_diversity_d10_d25_nr_mop$autoclave)
shapiro.test(alpha_diversity_d10_d25_nr_mop$q2)
ggqqplot(alpha_diversity_d10_d25_nr_mop,x="q2")

# Non-parametric alternative
wilcox.test(alpha_diversity_d10_d25_nr_mop$q2~alpha_diversity_d10_d25_nr_mop$autoclave)
wilcox_effsize(alpha_diversity_d10_d25_nr_mop,q2~autoclave)

# Converting to long
alpha_diversity_d10_d25_nr_mop_long<-gather(alpha_diversity_d10_d25_nr_mop,key="q",value="index",12:14)

# Creating vector of significance lettering based on Wilcox tests 
sig_code_autoclave<-data.frame(q=c("q0","q0","q1","q1","q2","q2"),lab=c("A","B","A","B","A","B"),x=c(1,2,1,2,1,2),y=c(975,975,125,125,40,40))

# Plotting in ggplot
autoclave_alpha_diversity<-ggplot(alpha_diversity_d10_d25_nr_mop_long,aes(x=autoclave,y=index,fill=autoclave))+
  geom_boxplot(outlier.shape = NA)+
  geom_jitter(inherit.aes=FALSE,aes(x=autoclave,y=index,fill=autoclave),shape=21,color="black",alpha=0.3)+
  facet_wrap(.~q,scales = "free_y")+
  geom_text(data=sig_code_autoclave,aes(x=x,y=y,label=lab,fontface="bold"),inherit.aes=FALSE)+
  guides(fill=guide_legend("Autoclave"))+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"))+
  ylab("Index Value")+
  xlab("Autoclave")+
  scale_fill_discrete(labels=c("Autoclaved","Non-Autoclaved"))+
  scale_color_discrete(labels=c("Autoclaved","Non-Autoclaved"))
  
  
autoclave_alpha_diversity

```


# Analyses of autoclaved soils alone (Day 10 and Day 25)

```{r}
# Subsetting ASV table to include only the Autoclave samples
seqtab_its_d10d25_nrmop_autoclave<-seqtab_its_d10d25_nrmop[grep("_A_",row.names(seqtab_its_d10d25_nrmop)),]

# Checking sample counts. All treatments should have 4 reps except for D25 NP_A which will have 6 reps. 
merge(sample.metadata,seqtab_its_d10d25_nrmop_autoclave,by.x=0,by.y=0) %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Generating rarefaction curves for all samples
rarefy.plot_its_d10d25_nrmop_autoclave<-rarecurve(seqtab_its_d10d25_nrmop_autoclave,tidy = TRUE)

# Plotting rarefaction plots in ggplot
rareplot_its_d10d25_nrmop_autoclave<-ggplot()+
  geom_line(data=rarefy.plot_its_d10d25_nrmop_autoclave,aes(x=Sample,y=Species,color=Site))+
  geom_vline(data=NULL,xintercept=c(1000,5000,10000,15000,20000,30000,40000))+
  theme(legend.position="none")+
  geom_text(data=NULL,aes(x=500,y=300,label="1000",angle=90))+
  geom_text(data=NULL,aes(x=4000,y=300,label="5000",angle=90))+
  geom_text(data=NULL,aes(x=9000,y=300,label="10000",angle=90))+
  geom_text(data=NULL,aes(x=14000,y=300,label="15000",angle=90))+
  geom_text(data=NULL,aes(x=19000,y=300,label="20000",angle=90))+
  geom_text(data=NULL,aes(x=29000,y=300,label="30000",angle=90))+
  geom_text(data=NULL,aes(x=39000,y=300,label="40000",angle=90))

rareplot_its_d10d25_nrmop_autoclave

sort(rowSums(seqtab_its_d10d25_nrmop_autoclave))

# Rarefying the ASV table to 10000 sequences
rare.asv.tab_its_d10d25_nrmop_autoclave<-as.data.frame(rrarefy(seqtab_its_d10d25_nrmop_autoclave, sample=1185))

# Removing samples that do not meet rarefaction cut-off and ASVs with zero reads following rarefaction.
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-rarefy_formatting(rare.asv.tab_its_d10d25_nrmop_autoclave,1185)

#write.table(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,"~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",row.names = TRUE,col.names=TRUE,sep = '\t')
```

# Comparisons of fungal communities within autoclaved soils only

```{r}
# loading in autoclaved rarefied ASV table 
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Loading in metadata
sample.metadata<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sample_metadata_from_sample_names.txt",header=TRUE,sep='\t',na.strings = "-")

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Beta-Diversity for all autoclaved samples regardless of sampling date
pcoa_its_within_autoclave<-pcoa_imager(metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,method = "bray",color="date",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_autoclave[[1]]
pcoa_its_within_autoclave_plot<-pcoa_its_within_autoclave[[2]]+
  guides(fill=guide_legend("Septoria Isolate"),color=guide_legend("Day"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_color_discrete(labels=c("Day 10","Day 25"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_within_autoclave_plot

# PERMANOVA
adonis2(Filter(x=metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,f=is.numeric)~metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$plant*metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$date+metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$septoria)

```

# Separating the autoclaved soil fungal communities by date starting with day 10. 

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Loading sample metadata
sample.metadata<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sample_metadata_from_sample_names.txt",header=TRUE,sep='\t',na.strings = "-")

# Loading taxonomy information
tax.tab_its<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/tax.tab_its.txt",header=TRUE,sep='\t')

row.names(tax.tab_its)<-sub(" ","\\.",row.names(tax.tab_its))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("D10",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$date),]

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10 %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Constructing relative abundance stacked bar charts at the family level. First taking the sum of all ASVs by sphaerulina x plant treatment
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10[,11:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10)]%>%
  as.data.frame()%>%
  group_by(sept_plant)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing row names to treatment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group$sept_plant

# Merging in the taxonomy information
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy<-merge(tax.tab_its,as.data.frame(t(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group)),by=0)

# Converting count data to numeric 
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy)]<-lapply(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy)],as.numeric)

# Taking the sum of each family
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy[,c(6,9:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy))]%>%
  as.data.frame()%>%
  group_by(Family)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing any NAs at the family level to unassigned
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy[which(is.na(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy$Family)),1]<-"Unassigned"

# Changing row names to the family assignment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy$Family
                                                                                
# Transposing the dataframe and taking family relative abundances by treatment
t.metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund<-as.data.frame(decostand(t(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy,f=is.numeric)),method='total'))

# Transposing the data frame again so that treatments are column names 
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund<-as.data.frame(t(t.metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund))

# Adding a column that is just the family names  
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund$family<-row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund)

# Subsetting the table so that only those ASVs with a total relative abundance greater than 0.1 are included
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_ten<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund[,1:6])>0.1,]

# Subsetting to create an other column that includes ASVs with a total relative abundance across treatment types less than 0.1
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_other<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund[,1:6])<0.1,]

# Adding a row to the data frame that is for the other/unassigned category
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_ten[nrow(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_ten)+1,1:6]<-colSums(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_other[,1:6])

# Changing the row names and family values for the other group
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_ten)[9]<-"Other"
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_ten[9,7]<-"Unassigned/Other"

# Converting table to long
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_long<-gather(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_ten,key="Treatment",value="Relative.Abundance",1:6)  

# Removing f__ label to clean names
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_long$family<-gsub("f__","",metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_long$family)

# Plotting relative abundance charts in ggplot
ggplot(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_long,aes(x=Treatment,y=Relative.Abundance))+
  geom_col(aes(x=Treatment,y=Relative.Abundance,fill=family),linewidth=1,inherit.aes = FALSE,color="black")+
 theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),legend.position = "bottom")+
  scale_x_discrete(labels=c("No Plant + Mn5","No Plant + Control","No Plant + SARM07","Plant + Mn5","Plant + Control","Plant + SARM07"))+
  guides(fill=guide_legend(title = "ASV ID"))+
  scale_fill_manual(values=RColorBrewer::brewer.pal(n=length(unique(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_group_taxonomy_relabund_long$family)),"Set3"))


# Beta-Diversity for all autoclaved samples regardless of sampling date
pcoa_its_within_autoclave_d10<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,method="bray",color="sept_plant",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_autoclave_d10[[1]]
pcoa_its_within_autoclave_d10_plot<-pcoa_its_within_autoclave_d10[[2]]+
  guides(fill=guide_legend("Plant Presence"),color=guide_legend("Sphaerulina Isolate x Plant Presence"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence) x No Plant","Control x No Plant","SARM07 (Low Virulence) x No Plant","Mn5 (High Virulence) x Plant","Control x Plant","SARM07 (Low Virulence) x Plant"))
pcoa_its_within_autoclave_d10_plot

# PERMANOVA
adonis2(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,f=is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10$plant*metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10$septoria)

# General Alpha-Diversity
alpha.diversity_nrmop_autoclave_d10<-data.frame(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,f=is.character),q0=hill_taxa(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,f=is.numeric),q=0),q1=hill_taxa(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,f=is.numeric),q=1),q2=hill_taxa(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,f=is.numeric),q=2))

# ANOVA q0
anova.alpha.diversity_autoclave_d10_q0<-aov(q0~plant*septoria,data = alpha.diversity_nrmop_autoclave_d10)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_d10_q0)
car::Anova(anova.alpha.diversity_autoclave_d10_q0,type="III")

# ANOVA q1
anova.alpha.diversity_autoclave_d10_q1<-aov(q1~plant*septoria,data = alpha.diversity_nrmop_autoclave_d10)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_d10_q1)
car::Anova(anova.alpha.diversity_autoclave_d10_q1,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_d10_q1)

# ANOVA q2
anova.alpha.diversity_autoclave_d10_q2<-aov(q2~plant*septoria,data = alpha.diversity_nrmop_autoclave_d10)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_d10_q2)
car::Anova(anova.alpha.diversity_autoclave_d10_q2,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_d10_q2)

# Converting to long format to plot in ggplot
alpha.diversity_nrmop_autoclave_d10_long<-gather(alpha.diversity_nrmop_autoclave_d10,key="q",value="index",12:14)

# Creating an object that contains the significance coding based on ANOVAs
sig_code_autoclave_d10<-data.frame(q=c("q0","q0","q1","q1","q1","q1","q1","q1","q2","q2","q2"),lab=c(NA,NA,"A","B","AB",NA,NA,NA,"A","B","AB"),x=c(1,2,0.7,1,1.3,1.6,2,2.3,0.7,1,1.3),y=c(150,150,11.5,11.5,11.5,11.5,11.5,11.5,8.3,8.3,8.3))

#xstart=c(NA,NA,0.60,1.60,0.60,1.60),xend=c(NA,NA,1.50,2.50,1.50,2.50),ystart=c(NA,NA,11,11,8,8),yend=c(NA,NA,11,11,8,8)

# Plotting in ggplot
alpha.diversity_its_nrmop_autoclave_d10_plot<-ggplot(alpha.diversity_nrmop_autoclave_d10_long,aes(x=plant,y=index,fill=septoria))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(inherit.aes=FALSE,aes(x=plant,y=index,fill=septoria),shape=21,color="black",alpha=0.9,position=position_jitterdodge(dodge.width = .90))+
  facet_wrap(.~q,scales = "free_y")+
  geom_text(data=sig_code_autoclave_d10,aes(x=x,y=y,label=lab,fontface="bold"),inherit.aes=FALSE)+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"))+
  scale_x_discrete(labels=c("No Plant","Plant"))+
  ylab("Index Value")+
  xlab("Plant Presence")+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  #geom_segment(data=sig_code_autoclave_d10,aes(x=xstart,xend=xend,y=ystart,yend=yend),inherit.aes=FALSE)+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"))
  
  
alpha.diversity_its_nrmop_autoclave_d10_plot
```

# PIME Analysis for Day 10 Autoclaved Soils

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Loading in sample metadata
sample.metadata<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sample_metadata_from_sample_names.txt",header=TRUE,sep='\t',na.strings = "-")

# Loading in taxonomy data
tax.tab_its<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/tax.tab_its.txt",header=TRUE,sep='\t')

# Removing periods from ASV labels
colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave)<-sub("\\."," ",colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("D10",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$date),]

# Adding row names to metadata file to allow for creation of phyloseq object
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10$Row.names

# Creating an OTU table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d10_otu_table<-phyloseq::otu_table(object=select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,is.numeric),taxa_are_rows = FALSE)

# Creating a taxonomy table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d10_tax_table<-phyloseq::tax_table(as.matrix(subset(tax.tab_its,row.names(tax.tab_its)%in%colnames(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10,is.numeric)))))

# Creating a metadata file for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d10_metadata<-phyloseq::sample_data(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10[,2:11])

# Creating phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d10_phyloseq_obj<-phyloseq::phyloseq(rare.asv.tab.fmt_its_nrmop_autoclave_d10_otu_table,rare.asv.tab.fmt_its_nrmop_autoclave_d10_tax_table,rare.asv.tab.fmt_its_nrmop_autoclave_d10_metadata)

# Estimating the out of bag error rate prior to prevalence filtering
pime.oob.error(rare.asv.tab.fmt_its_nrmop_autoclave_d10_phyloseq_obj,"sept_plant")

# Splitting data by sphaerulina and plant treatment
pime.variable.split_nrmop_autoclave_d10<-pime.split.by.variable(rare.asv.tab.fmt_its_nrmop_autoclave_d10_phyloseq_obj,"sept_plant")

# Creating prevalence intervals for each sphaerulina/plant treatment combination
pime.prevalence_nrmop_autoclave_d10<-pime.prevalence(pime.variable.split_nrmop_autoclave_d10)

# Calculating the random forest error rate for each possible prevalence (in increments of 5)
set.seed(42)
pime.best.prevalence_nrmop_autoclave_d10<-pime.best.prevalence(pime.prevalence_nrmop_autoclave_d10,"sept_plant")

# Pulling out the important ASVs for the model with a low error rate
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10<-as.data.frame(pime.best.prevalence_nrmop_autoclave_d10$Importance$`Prevalence 85`)

# Pulling out the phyloseq object for the model with a low error rate
pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d10<-pime.prevalence_nrmop_autoclave_d10$'85'

# Assessing model reliability by assigning treatments randomly to samples and computing error rate (expect that it should be an error rate greater than the model error rate when using the actual sample designations that does not change regardless of prevalence filtering). 
pime.randomized.model_nrmop_autoclave_d10<-pime.error.prediction(rare.asv.tab.fmt_its_nrmop_autoclave_d10_phyloseq_obj, "sept_plant", bootstrap = 100, parallel = TRUE, max.prev = 95)
pime.randomized.model_nrmop_autoclave_d10$Plot

# Assessing model error rates over multiple bootstrap replicates
replicated.oob.error_nrmop_autoclave_d10 <- pime.oob.replicate(pime.prevalence_nrmop_autoclave_d10, "sept_plant", bootstrap = 100, parallel = TRUE)
replicated.oob.error_nrmop_autoclave_d10$Plot

# Pulling out the ASV table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.asv.table_nrmop_autoclave_d10<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d10@otu_table)

# Pulling out the sample metadata from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.metadata_nrmop_autoclave_d10<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d10@sam_data)

# Pulling out the taxonomy table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.taxonomy_nrmop_autoclave_d10<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d10@tax_table)

# Removing the period from the ASV labels to match with the original sequence IDs
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$SequenceID<-sub("\\."," ",pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$SequenceID)

# Creating new labels to use for ASV importance plot that include the sequence id (ASV and number) and the family assignment
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$tax_id<-sub("f__","",paste0(pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$SequenceID,"_",pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$Family))

# Adjusting labels for the plot to make it look slightly more visually appealing.
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$tax_id<-sub("asv_its ","asv.its.",pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$tax_id)

# Creating new Sphaerulina object to match with the object created above
sphaerulina_asvs_named<-c(c("asv.its.360_Mycosphaerellaceae","asv.its.454_Mycosphaerellaceae","asv.its.1237_Mycosphaerellaceae","asv.its.1238_Mycosphaerellaceae","asv.its.1239_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae","asv.its.1249_Mycosphaerellaceae","asv.its.1288_Mycosphaerellaceae","asv.its.1352_Mycosphaerellaceae","asv.its.2539_Mycosphaerellaceae","asv.its.3962_Mycosphaerellaceae"))

key_sphaerulina<-c("asv.its.1237_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae")

# Plotting the ASV contributions to model error rates. This will highlight Sphaerulina ASVs in red
ggplot(data=pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10,aes(x=reorder(tax_id,MeanDecreaseAccuracy),y=MeanDecreaseAccuracy*100))+
  geom_col()+
  ylab("Mean Decrease in Model Accuracy (%)")+
  xlab("ASV")+
  coord_flip()+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),axis.text.y = element_text(color=rev(c(ifelse(pime.prevalence.filtered.important.asvs_nrmop_autoclave_d10$tax_id%in%key_sphaerulina,"red","black")))))

# Constructing relative abundance charts for total and core microbiome
pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d10<-merge(pime.prevalence.filtered.metadata_nrmop_autoclave_d10,pime.prevalence.filtered.asv.table_nrmop_autoclave_d10,by=0)

# Taking the sum of each taxon by the sphaerulina x plant treatment
pime.prevalence.filtered_nrmop_autoclave_d10_core<-pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d10[,c(11:length(pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d10))]%>%
  as.data.frame()%>%
  group_by(sept_plant)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing the rownames to the treatment ID
row.names(pime.prevalence.filtered_nrmop_autoclave_d10_core)<-pime.prevalence.filtered_nrmop_autoclave_d10_core$sept_plant

# Converting the taxon counts to per treatment relative abundance
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund<-as.data.frame(decostand(Filter(x=pime.prevalence.filtered_nrmop_autoclave_d10_core,f=is.numeric),method='total'))

# Merging in taxonomy information
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy<-merge(pime.prevalence.filtered.taxonomy_nrmop_autoclave_d10,t(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund),by=0)

# Subsetting so that only ASVs with a relative abundance greater than 0.1 across all treatments are included
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_ten<-pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy[rowSums(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy[,9:14])>0.1,]  

# Subsetting so that ASVs with a relative abundance less than 0.1 across all treatment are in their own object
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_other<-pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy[rowSums(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy[,9:14])<0.1,]

# Taking sum of low abundace taxa
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_ten[nrow(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_ten)+1,9:14]<-colSums(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_other[,9:14])

# Chaning row name of newly added other category
row.names(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_ten)[10]<-"Other"

# Chaning name for newly added other category
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_ten[10,]$Family<-"Unassigned/Other"

# Converting to long
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long<-gather(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_ten,key="Treatment",value="Relative.Abundance",9:14)     
                                                                                                  
# Changing ASV labels 
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$taxonomy<-paste0(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$Row.names,"_",ifelse(is.na(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$Family)==FALSE,pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$Family,pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$Order))

# Cleaning ASV label
pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$taxonomy<-gsub("NA_","",gsub("_[a-z]__","_",gsub("_its ",".its.",pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$taxonomy)))

# Plotting in ggplot
ggplot(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long,aes(x=Treatment,y=Relative.Abundance))+
  geom_col(aes(x=Treatment,y=Relative.Abundance,fill=taxonomy),color=c(ifelse(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$taxonomy%in%key_sphaerulina,"black",NA)),linewidth=0.75,inherit.aes = FALSE)+
 theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),legend.position = "bottom")+
  scale_x_discrete(labels=c("No Plant + Mn5","No Plant + Control","No Plant + SARM07","Plant + Mn5","Plant + Control","Plant + SARM07"))+
  guides(fill=guide_legend(title = "ASV ID"))+
    scale_fill_manual(values=RColorBrewer::brewer.pal(n=length(unique(pime.prevalence.filtered_nrmop_autoclave_d10_core_relabund_taxonomy_long$taxonomy)),"Set3")[c(2,3,1,5,4,6:8,10,9)])

# Removing ASVs identified as Sphaerulina by blast to assess effect on PCoA
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_no_sphaerulina<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10[,!(colnames(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10)%in%sphaerulina_asvs)]

# PCoA for autoclaved soils without sphaerulina
pcoa_its_within_autoclave_d10_no_sphaerulina<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_no_sphaerulina,method = "bray",color="sept_plant",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_autoclave_d10_no_sphaerulina[[1]]

pcoa_its_within_autoclave_d10_no_sphaerulina_plot<-pcoa_its_within_autoclave_d10_no_sphaerulina[[2]]+
  guides(fill=guide_legend("Plant Presence"),color=guide_legend("Sphaerulina Isolate x Plant Presence"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence) x No Plant","Control x No Plant","SARM07 (Low Virulence) x No Plant","Mn5 (High Virulence) x Plant","Control x Plant","SARM07 (Low Virulence) x Plant"))
pcoa_its_within_autoclave_d10_no_sphaerulina_plot

# PERMANOVA for autoclaved soils without Sphaerulina
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_no_sphaerulina,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_no_sphaerulina$septoria*metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d10_no_sphaerulina$plant)
```

# Separating the autoclaved soil fungal communities by date with day 25. 

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Loading sample metadata
sample.metadata<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sample_metadata_from_sample_names.txt",header=TRUE,sep='\t',na.strings = "-")

# Loading taxonomy information
tax.tab_its<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/tax.tab_its.txt",header=TRUE,sep='\t')

row.names(tax.tab_its)<-sub(" ","\\.",row.names(tax.tab_its))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("D25",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$date),]

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25 %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Constructing relative abundance stacked bar charts at the family level. First taking the sum of all ASVs by sphaerulina x plant treatment
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25[,11:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25)]%>%
  as.data.frame()%>%
  group_by(sept_plant)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing row names to treatment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group$sept_plant

# Merging in the taxonomy information
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy<-merge(tax.tab_its,as.data.frame(t(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group)),by=0)

# Converting count data to numeric 
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy)]<-lapply(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy)],as.numeric)

# Taking the sum of each family
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy[,c(6,9:length(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy))]%>%
  as.data.frame()%>%
  group_by(Family)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing any NAs at the family level to unassigned
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy[which(is.na(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy$Family)),1]<-"Unassigned"

# Changing row names to the family assignment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy$Family
                                                                                
# Transposing the dataframe and taking family relative abundances by treatment
t.metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund<-as.data.frame(decostand(t(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy,f=is.numeric)),method='total'))

# Transposing the data frame again so that treatments are column names 
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund<-as.data.frame(t(t.metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund))

# Adding a column that is just the family names  
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund$family<-row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund)

# Subsetting the table so that only those ASVs with a total relative abundance greater than 0.1 are included
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_ten<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund[,1:6])>0.1,]

# Subsetting to create an other column that includes ASVs with a total relative abundance across treatment types less than 0.1
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_other<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund[,1:6])<0.1,]

# Adding a row to the data frame that is for the other/unassigned category
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_ten[nrow(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_ten)+1,1:6]<-colSums(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_other[,1:6])

# Changing the row names and family values for the other group
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_ten)[9]<-"Other"
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_ten[9,7]<-"Unassigned/Other"

# Converting table to long
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_long<-gather(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_ten,key="Treatment",value="Relative.Abundance",1:6)  

# Removing f__ label to clean names
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_long$family<-gsub("f__","",metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_long$family)

# Plotting relative abundance charts in ggplot
ggplot(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_long,aes(x=Treatment,y=Relative.Abundance))+
  geom_col(aes(x=Treatment,y=Relative.Abundance,fill=family),linewidth=1,inherit.aes = FALSE,color="black")+
 theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),legend.position = "bottom")+
  scale_x_discrete(labels=c("No Plant + Mn5","No Plant + Control","No Plant + SARM07","Plant + Mn5","Plant + Control","Plant + SARM07"))+
  guides(fill=guide_legend(title = "ASV ID"))+
  scale_fill_manual(values=RColorBrewer::brewer.pal(n=length(unique(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_group_taxonomy_relabund_long$family)),"Set3"))


# Beta-Diversity for all autoclaved samples regardless of sampling date
pcoa_its_within_autoclave_d25<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,method="bray",color="sept_plant",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_autoclave_d25[[1]]
pcoa_its_within_autoclave_d25_plot<-pcoa_its_within_autoclave_d25[[2]]+
  guides(fill=guide_legend("Plant Presence"),color=guide_legend("Sphaerulina Isolate x Plant Presence"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence) x No Plant","Control x No Plant","SARM07 (Low Virulence) x No Plant","Mn5 (High Virulence) x Plant","Control x Plant","SARM07 (Low Virulence) x Plant"))
pcoa_its_within_autoclave_d25_plot

# PERMANOVA
adonis2(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,f=is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25$plant+metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25$septoria)

# General Alpha-Diversity
alpha.diversity_nrmop_autoclave_d25<-data.frame(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,f=is.character),q0=hill_taxa(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,f=is.numeric),q=0),q1=hill_taxa(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,f=is.numeric),q=1),q2=hill_taxa(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,f=is.numeric),q=2))

# ANOVA q0
anova.alpha.diversity_autoclave_d25_q0<-aov(q0~plant+septoria,data = alpha.diversity_nrmop_autoclave_d25)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_d25_q0)
car::Anova(anova.alpha.diversity_autoclave_d25_q0,type="III")

# ANOVA q1
anova.alpha.diversity_autoclave_d25_q1<-aov(q1~plant+septoria,data = alpha.diversity_nrmop_autoclave_d25)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_d25_q1)
car::Anova(anova.alpha.diversity_autoclave_d25_q1,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_d25_q1)

# ANOVA q2
anova.alpha.diversity_autoclave_d25_q2<-aov(q2~plant+septoria,data = alpha.diversity_nrmop_autoclave_d25)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_d25_q2)
car::Anova(anova.alpha.diversity_autoclave_d25_q2,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_d25_q2)

# Converting to long format to plot in ggplot
alpha.diversity_nrmop_autoclave_d25_long<-gather(alpha.diversity_nrmop_autoclave_d25,key="q",value="index",12:14)


# Plotting in ggplot
alpha.diversity_its_nrmop_autoclave_d25_plot<-ggplot(alpha.diversity_nrmop_autoclave_d25_long,aes(x=plant,y=index,fill=septoria))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(inherit.aes=FALSE,aes(x=plant,y=index,fill=septoria),shape=21,color="black",alpha=0.9,position=position_jitterdodge(dodge.width = .90))+
  facet_wrap(.~q,scales = "free_y")+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"))+
  scale_x_discrete(labels=c("No Plant","Plant"))+
  ylab("Index Value")+
  xlab("Plant Presence")+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"))
  
  
alpha.diversity_its_nrmop_autoclave_d25_plot
```


# PIME Analysis for Day 25 Autoclaved Soils

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Loading in sample metadata
sample.metadata<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/sample_metadata_from_sample_names.txt",header=TRUE,sep='\t',na.strings = "-")

# Loading in taxonomy data
tax.tab_its<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/tax.tab_its.txt",header=TRUE,sep='\t')

# Removing periods from ASV labels
colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave)<-sub("\\."," ",colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 25 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("D25",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$date),]

# Adding row names to metadata file to allow for creation of phyloseq object
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25$Row.names

# Creating an OTU table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d25_otu_table<-phyloseq::otu_table(object=select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,is.numeric),taxa_are_rows = FALSE)

# Creating a taxonomy table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d25_tax_table<-phyloseq::tax_table(as.matrix(subset(tax.tab_its,row.names(tax.tab_its)%in%colnames(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25,is.numeric)))))

# Creating a metadata file for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d25_metadata<-phyloseq::sample_data(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25[,2:11])

# Creating phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_d25_phyloseq_obj<-phyloseq::phyloseq(rare.asv.tab.fmt_its_nrmop_autoclave_d25_otu_table,rare.asv.tab.fmt_its_nrmop_autoclave_d25_tax_table,rare.asv.tab.fmt_its_nrmop_autoclave_d25_metadata)

# Estimating the out of bag error rate prior to prevalence filtering
pime.oob.error(rare.asv.tab.fmt_its_nrmop_autoclave_d25_phyloseq_obj,"sept_plant")

# Splitting data by sphaerulina and plant treatment
pime.variable.split_nrmop_autoclave_d25<-pime.split.by.variable(rare.asv.tab.fmt_its_nrmop_autoclave_d25_phyloseq_obj,"sept_plant")

# Creating prevalence intervals for each sphaerulina/plant treatment combination
pime.prevalence_nrmop_autoclave_d25<-pime.prevalence(pime.variable.split_nrmop_autoclave_d25)

# Calculating the random forest error rate for each possible prevalence (in increments of 5)
set.seed(42)
pime.best.prevalence_nrmop_autoclave_d25<-pime.best.prevalence(pime.prevalence_nrmop_autoclave_d25,"sept_plant")

# Pulling out the important ASVs for the model with a low error rate
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25<-as.data.frame(pime.best.prevalence_nrmop_autoclave_d25$Importance$`Prevalence 85`)

# Pulling out the phyloseq object for the model with a low error rate
pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d25<-pime.prevalence_nrmop_autoclave_d25$'85'

# Assessing model reliability by assigning treatments randomly to samples and computing error rate (expect that it should be an error rate greater than the model error rate when using the actual sample designations that does not change regardless of prevalence filtering). 
pime.randomized.model_nrmop_autoclave_d25<-pime.error.prediction(rare.asv.tab.fmt_its_nrmop_autoclave_d25_phyloseq_obj, "sept_plant", bootstrap = 100, parallel = TRUE, max.prev = 95)
pime.randomized.model_nrmop_autoclave_d25$Plot

# Assessing model error rates over multiple bootstrap replicates
replicated.oob.error_nrmop_autoclave_d25 <- pime.oob.replicate(pime.prevalence_nrmop_autoclave_d25, "sept_plant", bootstrap = 100, parallel = TRUE)
replicated.oob.error_nrmop_autoclave_d25$Plot

# Pulling out the ASV table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.asv.table_nrmop_autoclave_d25<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d25@otu_table)

# Pulling out the sample metadata from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.metadata_nrmop_autoclave_d25<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d25@sam_data)

# Pulling out the taxonomy table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.taxonomy_nrmop_autoclave_d25<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_d25@tax_table)

# Removing the period from the ASV labels to match with the original sequence IDs
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$SequenceID<-sub("\\."," ",pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$SequenceID)

# Creating new labels to use for ASV importance plot that include the sequence id (ASV and number) and the family assignment
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$tax_id<-sub("f__","",paste0(pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$SequenceID,"_",pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$Family))

# Adjusting labels for the plot to make it look slightly more visually appealing.
pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$tax_id<-sub("asv_its ","asv.its.",pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$tax_id)

# Creating new Sphaerulina object to match with the object created above
sphaerulina_asvs_named<-c(c("asv.its.360_Mycosphaerellaceae","asv.its.454_Mycosphaerellaceae","asv.its.1237_Mycosphaerellaceae","asv.its.1238_Mycosphaerellaceae","asv.its.1239_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae","asv.its.1249_Mycosphaerellaceae","asv.its.1288_Mycosphaerellaceae","asv.its.1352_Mycosphaerellaceae","asv.its.2539_Mycosphaerellaceae","asv.its.3962_Mycosphaerellaceae"))

key_sphaerulina<-c("asv.its.1237_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae")

# Plotting the ASV contributions to model error rates. This will highlight Sphaerulina ASVs in red
ggplot(data=pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25,aes(x=reorder(tax_id,MeanDecreaseAccuracy),y=MeanDecreaseAccuracy*100))+
  geom_col()+
  ylab("Mean Decrease in Model Accuracy (%)")+
  xlab("ASV")+
  coord_flip()+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),axis.text.y = element_text(color=rev(c(ifelse(pime.prevalence.filtered.important.asvs_nrmop_autoclave_d25$tax_id%in%key_sphaerulina,"red","black")))))

# Constructing relative abundance charts for total and core microbiome
pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25<-merge(pime.prevalence.filtered.metadata_nrmop_autoclave_d25,pime.prevalence.filtered.asv.table_nrmop_autoclave_d25,by=0)

# Taking the sum of each taxon by the sphaerulina x plant treatment
pime.prevalence.filtered_nrmop_autoclave_d25_core<-pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25[,c(11:length(pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25))]%>%
  as.data.frame()%>%
  group_by(sept_plant)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing the rownames to the treatment ID
row.names(pime.prevalence.filtered_nrmop_autoclave_d25_core)<-pime.prevalence.filtered_nrmop_autoclave_d25_core$sept_plant

# Converting the taxon counts to per treatment relative abundance
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund<-as.data.frame(decostand(Filter(x=pime.prevalence.filtered_nrmop_autoclave_d25_core,f=is.numeric),method='total'))

# Merging in taxonomy information
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy<-merge(pime.prevalence.filtered.taxonomy_nrmop_autoclave_d25,t(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund),by=0)

# Subsetting so that only ASVs with a relative abundance greater than 0.1 across all treatments are included
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten<-pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy[rowSums(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy[,9:14])>0.1,]  

# Subsetting so that ASVs with a relative abundance less than 0.1 across all treatment are in their own object
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other<-pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy[rowSums(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy[,9:14])<0.1,]

pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other_myco<-pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other[grep("Mycosphaerellaceae",pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other$Family),]

pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other<-pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other[grep("Mycosphaerellaceae",pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other$Family,invert = TRUE),]

# Taking sum of low abundance taxa
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten[nrow(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten)+1,]<-pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other_myco

# Taking sum of low abundance taxa
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten[nrow(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten)+1,9:14]<-colSums(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_other[,9:14])

# Changing row name of newly added other category
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten$Row.names[9]<-"Other"

# Changing row names 
row.names(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten)<-pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten$Row.names

# Changing name for newly added other category
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten[9,]$Family<-"Unassigned/Other"

# Converting to long
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long<-gather(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_ten,key="Treatment",value="Relative.Abundance",9:14)     
                                                                                                  
# Changing ASV labels 
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$taxonomy<-paste0(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$Row.names,"_",ifelse(is.na(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$Family)==FALSE,pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$Family,pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$Order))

# Cleaning ASV label
pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$taxonomy<-gsub("NA_","",gsub("_[a-z]__","_",gsub("_its ",".its.",pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$taxonomy)))

# Plotting in ggplot
ggplot(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long,aes(x=Treatment,y=Relative.Abundance))+
  geom_col(aes(x=Treatment,y=Relative.Abundance,fill=taxonomy),color=c(ifelse(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$taxonomy%in%key_sphaerulina,"black",NA)),linewidth=0.75,inherit.aes = FALSE)+
 theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),legend.position = "bottom")+
  scale_x_discrete(labels=c("No Plant + Mn5","No Plant + Control","No Plant + SARM07","Plant + Mn5","Plant + Control","Plant + SARM07"))+
  guides(fill=guide_legend(title = "ASV ID"))+
    scale_fill_manual(values=c(RColorBrewer::brewer.pal(n=length(unique(pime.prevalence.filtered_nrmop_autoclave_d25_core_relabund_taxonomy_long$taxonomy)),"Set3")[c(2,3,1,5,4,6,7,8)],"#D9D9D9"))

#
pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25_no_sphaerulina<-pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25[,!(colnames(pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25)%in%sphaerulina_asvs)]

pcoa_its_within_autoclave_d25_core<-pcoa_imager(pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25_no_sphaerulina,method = "bray",color="sept_plant",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_autoclave_d25_core[[1]]

pcoa_its_within_autoclave_d25_core<-pcoa_its_within_autoclave_d25_core[[2]]+
  guides(fill=guide_legend("Plant Presence"),color=guide_legend("Sphaerulina Isolate x Plant Presence"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence) x No Plant","Control x No Plant","SARM07 (Low Virulence) x No Plant","Mn5 (High Virulence) x Plant","Control x Plant","SARM07 (Low Virulence) x Plant"))
pcoa_its_within_autoclave_d25_core

# PERMANOVA for autoclaved soils without Sphaerulina
adonis2(select_if(pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25_no_sphaerulina,is.numeric)~pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25_no_sphaerulina$septoria+pime.prevalence.filtered.metadata.asv.table_nrmop_autoclave_d25_no_sphaerulina$plant)

# Removing ASVs identified as Sphaerulina by blast to assess effect on PCoA
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_no_sphaerulina<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25[,!(colnames(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25)%in%sphaerulina_asvs)]

# PCoA for autoclaved soils without sphaerulina
pcoa_its_within_autoclave_d25_no_sphaerulina<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_no_sphaerulina,method = "bray",color="sept_plant",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_autoclave_d25_no_sphaerulina[[1]]

pcoa_its_within_autoclave_d25_no_sphaerulina_plot<-pcoa_its_within_autoclave_d25_no_sphaerulina[[2]]+
  guides(fill=guide_legend("Plant Presence"),color=guide_legend("Sphaerulina Isolate x Plant Presence"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence) x No Plant","Control x No Plant","SARM07 (Low Virulence) x No Plant","Mn5 (High Virulence) x Plant","Control x Plant","SARM07 (Low Virulence) x Plant"))
pcoa_its_within_autoclave_d25_no_sphaerulina_plot

# PERMANOVA for autoclaved soils without Sphaerulina
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_no_sphaerulina,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_no_sphaerulina$septoria*metadata.rare.asv.tab.fmt_its_nrmop_autoclave_d25_no_sphaerulina$plant)
```

# Separating the autoclaved soil fungal communities by plant presence starting without plant 

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("NP",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$plant),]

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Beta-Diversity for all autoclaved samples regardless of sampling date
pcoa_its_within_autoclave_np<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,method="bray",color="date_septoria",shape="septoria",polygon=TRUE,color_two="date")
pcoa_its_within_autoclave_np[[1]]
pcoa_its_within_autoclave_np_plot<-pcoa_its_within_autoclave_np[[2]]+
  guides(fill=guide_legend("Date"),color=guide_legend("Date x Sphaerulina Isolate"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("Day 10","Day 25"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Day 10 x Mn5 (High Virulence) ","Day 10 x Control","Day 10 x SARM07 (Low Virulence)","Day 25 x Mn5 (High Virulence)","Day 25 x Control","Day 25 x SARM07 (Low Virulence)"))
pcoa_its_within_autoclave_np_plot

# PERMANOVA
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np$date+metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np$septoria)

# General Alpha-Diversity
alpha.diversity_nrmop_autoclave_np<-data.frame(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,is.character),q0=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,is.numeric),q=0),q1=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,is.numeric),q=1),q2=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,is.numeric),q=2))

# ANOVA q0
anova.alpha.diversity_autoclave_np_q0<-aov(q0~date+septoria,data = alpha.diversity_nrmop_autoclave_np)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_np_q0)
car::Anova(anova.alpha.diversity_autoclave_np_q0,type="III")

kruskal.test(q0~septoria,alpha.diversity_nrmop_autoclave_np)

# ANOVA q1
anova.alpha.diversity_autoclave_np_q1<-aov(q1~date+septoria,data = alpha.diversity_nrmop_autoclave_np)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_np_q1)
car::Anova(anova.alpha.diversity_autoclave_np_q1,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_np_q1)

kruskal.test(q1~septoria,alpha.diversity_nrmop_autoclave_np)
dunnTest(q1~septoria,alpha.diversity_nrmop_autoclave_np)

# ANOVA q2
anova.alpha.diversity_autoclave_np_q2<-aov(q2~date+septoria,data = alpha.diversity_nrmop_autoclave_np)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_np_q2)
car::Anova(anova.alpha.diversity_autoclave_np_q2,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_np_q2)

kruskal.test(q2~septoria,alpha.diversity_nrmop_autoclave_np)
dunnTest(q2~septoria,alpha.diversity_nrmop_autoclave_np)

# Converting to long format to plot in ggplot
alpha.diversity_nrmop_autoclave_d10_long<-gather(alpha.diversity_nrmop_autoclave_np,key="q",value="index",12:14)

# Creating an object that contains the significance coding based on ANOVAs
sig_code_autoclave_np<-data.frame(q=c("q0","q0","q0","q1","q1","q1","q2","q2","q2"),lab=c(NA,NA,NA,"A","B","AB","A","B","AB"),x=c(1,2,3,1,2,3,1,2,3),y=c(60,60,60,15,15,15,9,9,9))


# Plotting in ggplot
alpha.diversity_its_nrmop_autoclave_np_plot<-ggplot(alpha.diversity_nrmop_autoclave_d10_long,aes(x=septoria,y=index,fill=septoria))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(inherit.aes=FALSE,aes(x=septoria,y=index,fill=septoria,shape=date),color="black",alpha=0.9,position=position_jitterdodge(dodge.width = .90),size=2)+
  scale_shape_manual(values=c(21,22),labels=c("Day 10","Day 25"))+
  facet_wrap(.~q,scales = "free_y")+
  geom_text(data=sig_code_autoclave_np,aes(x=x,y=y,label=lab,fontface="bold"),inherit.aes=FALSE)+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"))+
  scale_x_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  ylab("Index Value")+
  xlab("Sphaerulina Isolate")+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"),shape=guide_legend("Collection Date"))
  
  
alpha.diversity_its_nrmop_autoclave_np_plot
```


# PIME Analysis for No plant Autoclaved Soils

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Removing periods from ASV labels
colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave)<-sub("\\."," ",colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("NP",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$plant),]

# Adding row names to metadata file to allow for creation of phyloseq object
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np$Row.names

# Creating an OTU table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_np_otu_table<-phyloseq::otu_table(object=select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,is.numeric),taxa_are_rows = FALSE)

# Creating a taxonomy table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_np_tax_table<-phyloseq::tax_table(as.matrix(subset(tax.tab_its,row.names(tax.tab_its)%in%colnames(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np,is.numeric)))))

# Creating a metadata file for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_np_metadata<-phyloseq::sample_data(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np[,2:11])

# Creating phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_np_phyloseq_obj<-phyloseq::phyloseq(rare.asv.tab.fmt_its_nrmop_autoclave_np_otu_table,rare.asv.tab.fmt_its_nrmop_autoclave_np_tax_table,rare.asv.tab.fmt_its_nrmop_autoclave_np_metadata)

# Estimating the out of bag error rate prior to prevalence filtering
pime.oob.error(rare.asv.tab.fmt_its_nrmop_autoclave_np_phyloseq_obj,"sept_plant")

# Splitting data by sphaerulina and plant treatment
pime.variable.split_nrmop_autoclave_np<-pime.split.by.variable(rare.asv.tab.fmt_its_nrmop_autoclave_np_phyloseq_obj,"date_septoria")

# Creating prevalence intervals for each sphaerulina/plant treatment combination
pime.prevalence_nrmop_autoclave_np<-pime.prevalence(pime.variable.split_nrmop_autoclave_np)

# Calculating the random forest error rate for each possible prevalence (in increments of 5)
set.seed(42)
pime.best.prevalence_nrmop_autoclave_np<-pime.best.prevalence(pime.prevalence_nrmop_autoclave_np,"date_septoria")

# Pulling out the important ASVs for the model with a low error rate
pime.prevalence.filtered.important.asvs_nrmop_autoclave_np<-as.data.frame(pime.best.prevalence_nrmop_autoclave_np$Importance$`Prevalence 85`)

# Pulling out the phyloseq object for the model with a low error rate
pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_np<-pime.prevalence_nrmop_autoclave_np$'85'

# Assessing model reliability by assigning treatments randomly to samples and computing error rate (expect that it should be an error rate greater than the model error rate when using the actual sample designations that does not change regardless of prevalence filtering). 
pime.randomized.model_nrmop_autoclave_np<-pime.error.prediction(rare.asv.tab.fmt_its_nrmop_autoclave_np_phyloseq_obj, "date_septoria", bootstrap = 100, parallel = TRUE, max.prev = 95)
pime.randomized.model_nrmop_autoclave_np$Plot

# Assessing model error rates over multiple bootstrap replicates
replicated.oob.error_nrmop_autoclave_np <- pime.oob.replicate(pime.prevalence_nrmop_autoclave_np, "date_septoria", bootstrap = 100, parallel = TRUE)
replicated.oob.error_nrmop_autoclave_np$Plot

# Pulling out the ASV table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.asv.table_nrmop_autoclave_np<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_np@otu_table)

# Pulling out the sample metadata from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.metadata_nrmop_autoclave_np<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_np@sam_data)

# Pulling out the taxonomy table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.taxonomy_nrmop_autoclave_np<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_np@tax_table)

# Removing the period from the ASV labels to match with the original sequence IDs
pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$SequenceID<-sub("\\."," ",pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$SequenceID)

# Creating new labels to use for ASV importance plot that include the sequence id (ASV and number) and the family assignment
pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$tax_id<-sub("f__","",paste0(pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$SequenceID,"_",pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$Family))

# Adjusting labels for the plot to make it look slightly more visually appealing.
pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$tax_id<-sub("asv_its ","asv.its.",pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$tax_id)

# Creating new Sphaerulina object to match with the object created above
sphaerulina_asvs_named<-c(c("asv.its.360_Mycosphaerellaceae","asv.its.454_Mycosphaerellaceae","asv.its.1237_Mycosphaerellaceae","asv.its.1238_Mycosphaerellaceae","asv.its.1239_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae","asv.its.1249_Mycosphaerellaceae","asv.its.1288_Mycosphaerellaceae","asv.its.1352_Mycosphaerellaceae","asv.its.2539_Mycosphaerellaceae","asv.its.3962_Mycosphaerellaceae"))

# Plotting the ASV contributions to model error rates. This will highlight Sphaerulina ASVs in red
ggplot(data=pime.prevalence.filtered.important.asvs_nrmop_autoclave_np,aes(x=reorder(tax_id,MeanDecreaseAccuracy),y=MeanDecreaseAccuracy*100))+
  geom_col()+
  ylab("Mean Decrease in Model Accuracy (%)")+
  xlab("ASV")+
  coord_flip()+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),axis.text.y = element_text(color=rev(c(ifelse(pime.prevalence.filtered.important.asvs_nrmop_autoclave_np$tax_id%in%sphaerulina_asvs_named,"red","black")))))

# Removing ASVs identified as Sphaerulina by blast to assess effect on PCoA
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np_no_sphaerulina<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np[,!(colnames(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np)%in%sphaerulina_asvs)]

# PCoA for autoclaved soils without sphaerulina
pcoa_its_within_autoclave_np_no_sphaerulina<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np_no_sphaerulina,method="bray",color="date_septoria",shape="septoria",polygon=TRUE,color_two="date")
pcoa_its_within_autoclave_np_no_sphaerulina[[1]]
pcoa_its_within_autoclave_np_no_sphaerulina_plot<-pcoa_its_within_autoclave_np_no_sphaerulina[[2]]+
  guides(fill=guide_legend("Date"),color=guide_legend("Date x Sphaerulina Isolate"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("Day 10","Day 25"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Day 10 x Mn5 (High Virulence) ","Day 10 x Control","Day 10 x SARM07 (Low Virulence)","Day 25 x Mn5 (High Virulence)","Day 25 x Control","Day 25 x SARM07 (Low Virulence)"))
pcoa_its_within_autoclave_np_no_sphaerulina_plot

# PERMANOVA
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np_no_sphaerulina,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np_no_sphaerulina$date+metadata.rare.asv.tab.fmt_its_nrmop_autoclave_np_no_sphaerulina$septoria)
```


# Separating the autoclaved soil fungal communities by plant presence starting with plant

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("NP",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$plant,invert = TRUE),]

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Beta-Diversity for all autoclaved samples regardless of sampling date
pcoa_its_within_autoclave_p<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,method="bray",color="date_septoria",shape="septoria",polygon=TRUE,color_two="date")
pcoa_its_within_autoclave_p[[1]]
pcoa_its_within_autoclave_p_plot<-pcoa_its_within_autoclave_p[[2]]+
  guides(fill=guide_legend("Date"),color=guide_legend("Date x Sphaerulina Isolate"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("Day 10","Day 25"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Day 10 x Mn5 (High Virulence) ","Day 10 x Control","Day 10 x SARM07 (Low Virulence)","Day 25 x Mn5 (High Virulence)","Day 25 x Control","Day 25 x SARM07 (Low Virulence)"))
pcoa_its_within_autoclave_p_plot

# PERMANOVA
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p$date*metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p$septoria)

# General Alpha-Diversity
alpha.diversity_nrmop_autoclave_p<-data.frame(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,is.character),q0=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,is.numeric),q=0),q1=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,is.numeric),q=1),q2=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,is.numeric),q=2))

# ANOVA q0
anova.alpha.diversity_autoclave_p_q0<-aov(q0~date*septoria,data = alpha.diversity_nrmop_autoclave_p)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_p_q0)
car::Anova(anova.alpha.diversity_autoclave_p_q0,type="III")

kruskal.test(q0~septoria,alpha.diversity_nrmop_autoclave_p)

# ANOVA q1
anova.alpha.diversity_autoclave_p_q1<-aov(q1~date*septoria,data = alpha.diversity_nrmop_autoclave_p)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_p_q1)
car::Anova(anova.alpha.diversity_autoclave_p_q1,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_p_q1)

kruskal.test(q1~septoria,alpha.diversity_nrmop_autoclave_p)
dunnTest(q1~septoria,alpha.diversity_nrmop_autoclave_p)

# ANOVA q2
anova.alpha.diversity_autoclave_p_q2<-aov(q2~date*septoria,data = alpha.diversity_nrmop_autoclave_p)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_autoclave_p_q2)
car::Anova(anova.alpha.diversity_autoclave_p_q2,type="III")
TukeyHSD(anova.alpha.diversity_autoclave_p_q2)

kruskal.test(q2~septoria,alpha.diversity_nrmop_autoclave_p)
dunnTest(q2~septoria,alpha.diversity_nrmop_autoclave_p)

# Converting to long format to plot in ggplot
alpha.diversity_nrmop_autoclave_d10_long<-gather(alpha.diversity_nrmop_autoclave_p,key="q",value="index",12:14)

# Plotting in ggplot
alpha.diversity_its_nrmop_autoclave_p_plot<-ggplot(alpha.diversity_nrmop_autoclave_d10_long,aes(x=septoria,y=index,fill=septoria))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(inherit.aes=FALSE,aes(x=septoria,y=index,fill=septoria,shape=date),color="black",alpha=0.9,position=position_jitterdodge(dodge.width = .90),size=2)+
  scale_shape_manual(values=c(21,22),labels=c("Day 10","Day 25"))+
  facet_wrap(.~q,scales = "free_y")+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"))+
  scale_x_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  ylab("Index Value")+
  xlab("Sphaerulina Isolate")+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"),shape=guide_legend("Collection Date"))
  
  
alpha.diversity_its_nrmop_autoclave_p_plot
```


# PIME Analysis for plant Autoclaved Soils

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_autoclave.txt",header=TRUE,sep='\t')

# Removing periods from ASV labels
colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave)<-sub("\\."," ",colnames(rare.asv.tab.fmt_its_d10d25_nrmop_autoclave))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_autoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave[grep("NP",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_autoclave$plant,invert=TRUE),]

# Adding row names to metadata file to allow for creation of phyloseq object
row.names(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p)<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p$Row.names

# Creating an OTU table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_p_otu_table<-phyloseq::otu_table(object=select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,is.numeric),taxa_are_rows = FALSE)

# Creating a taxonomy table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_p_tax_table<-phyloseq::tax_table(as.matrix(subset(tax.tab_its,row.names(tax.tab_its)%in%colnames(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p,is.numeric)))))

# Creating a metadata file for the phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_p_metadata<-phyloseq::sample_data(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p[,2:11])

# Creating phyloseq object
rare.asv.tab.fmt_its_nrmop_autoclave_p_phyloseq_obj<-phyloseq::phyloseq(rare.asv.tab.fmt_its_nrmop_autoclave_p_otu_table,rare.asv.tab.fmt_its_nrmop_autoclave_p_tax_table,rare.asv.tab.fmt_its_nrmop_autoclave_p_metadata)

# Estimating the out of bag error rate prior to prevalence filtering
pime.oob.error(rare.asv.tab.fmt_its_nrmop_autoclave_p_phyloseq_obj,"date_septoria")

# Splitting data by sphaerulina and plant treatment
pime.variable.split_nrmop_autoclave_p<-pime.split.by.variable(rare.asv.tab.fmt_its_nrmop_autoclave_p_phyloseq_obj,"date_septoria")

# Creating prevalence intervals for each sphaerulina/plant treatment combination
pime.prevalence_nrmop_autoclave_p<-pime.prevalence(pime.variable.split_nrmop_autoclave_p)

# Calculating the random forest error rate for each possible prevalence (in increments of 5)
set.seed(42)
pime.best.prevalence_nrmop_autoclave_p<-pime.best.prevalence(pime.prevalence_nrmop_autoclave_p,"date_septoria")

# Pulling out the important ASVs for the model with a low error rate
pime.prevalence.filtered.important.asvs_nrmop_autoclave_p<-as.data.frame(pime.best.prevalence_nrmop_autoclave_p$Importance$`Prevalence 85`)

# Pulling out the phyloseq object for the model with a low error rate
pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_p<-pime.prevalence_nrmop_autoclave_p$'85'

# Assessing model reliability by assigning treatments randomly to samples and computing error rate (expect that it should be an error rate greater than the model error rate when using the actual sample designations that does not change regardless of prevalence filtering). 
pime.randomized.model_nrmop_autoclave_p<-pime.error.prediction(rare.asv.tab.fmt_its_nrmop_autoclave_p_phyloseq_obj, "date_septoria", bootstrap = 100, parallel = TRUE, max.prev = 95)
pime.randomized.model_nrmop_autoclave_p$Plot

# Assessing model error rates over multiple bootstrap replicates
replicated.oob.error_nrmop_autoclave_p <- pime.oob.replicate(pime.prevalence_nrmop_autoclave_p, "date_septoria", bootstrap = 100, parallel = TRUE)
replicated.oob.error_nrmop_autoclave_p$Plot

# Pulling out the ASV table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.asv.table_nrmop_autoclave_p<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_p@otu_table)

# Pulling out the sample metadata from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.metadata_nrmop_autoclave_p<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_p@sam_data)

# Pulling out the taxonomy table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.taxonomy_nrmop_autoclave_p<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_autoclave_p@tax_table)

# Removing the period from the ASV labels to match with the original sequence IDs
pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$SequenceID<-sub("\\."," ",pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$SequenceID)

# Creating new labels to use for ASV importance plot that include the sequence id (ASV and number) and the family assignment
pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$tax_id<-sub("f__","",paste0(pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$SequenceID,"_",pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$Family))

# Adjusting labels for the plot to make it look slightly more visually appealing.
pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$tax_id<-sub("asv_its ","asv.its.",pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$tax_id)

# Removing negative values
pime.prevalence.filtered.important.asvs_nrmop_autoclave_p_filt<-pime.prevalence.filtered.important.asvs_nrmop_autoclave_p[pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$MeanDecreaseAccuracy>0,]

# Creating new Sphaerulina object to match with the object created above
sphaerulina_asvs_named<-c(c("asv.its.360_Mycosphaerellaceae","asv.its.454_Mycosphaerellaceae","asv.its.1237_Mycosphaerellaceae","asv.its.1238_Mycosphaerellaceae","asv.its.1239_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae","asv.its.1249_Mycosphaerellaceae","asv.its.1288_Mycosphaerellaceae","asv.its.1352_Mycosphaerellaceae","asv.its.2539_Mycosphaerellaceae","asv.its.3962_Mycosphaerellaceae"))

# Plotting the ASV contributions to model error rates. This will highlight Sphaerulina ASVs in red
ggplot(data=pime.prevalence.filtered.important.asvs_nrmop_autoclave_p_filt,aes(x=reorder(tax_id,MeanDecreaseAccuracy),y=MeanDecreaseAccuracy*100))+
  geom_col()+
  ylab("Mean Decrease in Model Accuracy (%)")+
  xlab("ASV")+
  coord_flip()+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),axis.text.y = element_text(color=rev(c(ifelse(pime.prevalence.filtered.important.asvs_nrmop_autoclave_p$tax_id%in%sphaerulina_asvs_named,"red","black")))))

# Removing ASVs identified as Sphaerulina by blast to assess effect on PCoA
metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p_no_sphaerulina<-metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p[,!(colnames(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p)%in%sphaerulina_asvs)]

# PCoA for autoclaved soils without sphaerulina
pcoa_its_within_autoclave_p_no_sphaerulina<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p_no_sphaerulina,method="bray",color="date_septoria",shape="septoria",polygon=TRUE,color_two="date")
pcoa_its_within_autoclave_p_no_sphaerulina[[1]]
pcoa_its_within_autoclave_p_no_sphaerulina_plot<-pcoa_its_within_autoclave_p_no_sphaerulina[[2]]+
  guides(fill=guide_legend("Date"),color=guide_legend("Date x Sphaerulina Isolate"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("Day 10","Day 25"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Day 10 x Mn5 (High Virulence) ","Day 10 x Control","Day 10 x SARM07 (Low Virulence)","Day 25 x Mn5 (High Virulence)","Day 25 x Control","Day 25 x SARM07 (Low Virulence)"))
pcoa_its_within_autoclave_p_no_sphaerulina_plot

# PERMANOVA
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p_no_sphaerulina,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p_no_sphaerulina$date*metadata.rare.asv.tab.fmt_its_nrmop_autoclave_p_no_sphaerulina$septoria)
```

# Analyses of nonautoclaved soils alone (Day 10 and Day 25)

```{r}
# Subsetting ASV table to include only the Autoclave samples
seqtab_its_d10d25_nrmop_nonautoclave<-seqtab_its_d10d25_nrmop[grep("_NA_",row.names(seqtab_its_d10d25_nrmop)),]

# Checking sample counts. All treatments should have 4 reps except for D25 NP_A which will have 6 reps. 
merge(sample.metadata,seqtab_its_d10d25_nrmop_nonautoclave,by.x=0,by.y=0) %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Generating rarefaction curves for all samples
rarefy.plot_its_d10d25_nrmop_nonautoclave<-rarecurve(seqtab_its_d10d25_nrmop_nonautoclave,tidy = TRUE)

# Plotting rarefaction plots in ggplot
rareplot_its_d10d25_nrmop_autoclave<-ggplot()+
  geom_line(data=rarefy.plot_its_d10d25_nrmop_autoclave,aes(x=Sample,y=Species,color=Site))+
  geom_vline(data=NULL,xintercept=c(1000,5000,10000,15000,20000,30000,40000))+
  theme(legend.position="none")+
  geom_text(data=NULL,aes(x=500,y=300,label="1000",angle=90))+
  geom_text(data=NULL,aes(x=4000,y=300,label="5000",angle=90))+
  geom_text(data=NULL,aes(x=9000,y=300,label="10000",angle=90))+
  geom_text(data=NULL,aes(x=14000,y=300,label="15000",angle=90))+
  geom_text(data=NULL,aes(x=19000,y=300,label="20000",angle=90))+
  geom_text(data=NULL,aes(x=29000,y=300,label="30000",angle=90))+
  geom_text(data=NULL,aes(x=39000,y=300,label="40000",angle=90))

rareplot_its_d10d25_nrmop_nonautoclave

sort(rowSums(seqtab_its_d10d25_nrmop_nonautoclave))

# Rarefying the ASV table to 10000 sequences
rare.asv.tab_its_d10d25_nrmop_nonautoclave<-as.data.frame(rrarefy(seqtab_its_d10d25_nrmop_nonautoclave, sample=21056))

# Removing samples that do not meet rarefaction cut-off and ASVs with zero reads following rarefaction.
rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-rarefy_formatting(rare.asv.tab_its_d10d25_nrmop_nonautoclave,21056)

#write.table(rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,"~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave.txt",row.names = TRUE,col.names=TRUE,sep = '\t')
```


# Funguild Analyses

```{r}
# Importing Funguild Output
funguild_soil<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/funguild_input_formatted_soil_asv_table.guilds.txt",header=TRUE,sep='\t')

# Subsetting Funguild Output to remove ASV count information
funguild_soil_assignments_only<-funguild_soil[,c(1,291:300)]

# Pulling out only funguild assignments
funguild_soil_assignments_only_possible_guild<-funguild_soil_assignments_only[,c(1,3,6)]

# Reformatting output to change unassigned values to - and separators in assigned ASVs to _
funguild_soil_assignments_only_possible_guild$Guild<-ifelse(grepl("[A-Z]",funguild_soil_assignments_only_possible_guild$Guild)==FALSE,gsub("-","NA",funguild_soil_assignments_only_possible_guild$Guild,fixed = TRUE), gsub("-","_",funguild_soil_assignments_only_possible_guild$Guild))
```


# Comparisons of fungal communities within autoclaved soils only

```{r}
# loading in autoclaved rarefied ASV table 
rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave.txt",header=TRUE,sep='\t')

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,by.x=0,by.y=0)

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Beta-Diversity for all autoclaved samples regardless of sampling date
pcoa_its_within_nonautoclave<-pcoa_imager(metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,method = "bray",color="date",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_nonautoclave[[1]]
pcoa_its_within_nonautoclave_plot<-pcoa_its_within_nonautoclave[[2]]+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Day"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_color_discrete(labels=c("Day 10","Day 25"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_within_nonautoclave_plot

# PERMANOVA
adonis2(select_if(metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,is.numeric)~metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave$plant*metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave$date*metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave$septoria)

```

# Separating the nonautoclaved soil fungal communities by date starting with day 10. 

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave.txt",header=TRUE,sep='\t')

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave[grep("D10",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave$date),]

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10 %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Constructing relative abundance stacked bar charts at the family level. First taking the sum of all ASVs by sphaerulina x plant treatment
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10[,11:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10)]%>%
  as.data.frame()%>%
  group_by(sept_plant)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing row names to treatment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group)<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group$sept_plant
colnames(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group)<-sub("\\."," ",colnames(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group))

# Merging in the taxonomy information
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy<-merge(tax.tab_its,as.data.frame(t(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group)),by=0)

# Converting count data to numeric 
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy)]<-lapply(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy)],as.numeric)

# Taking the sum of each family
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy[,c(6,9:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy))]%>%
  as.data.frame()%>%
  group_by(Family)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing any NAs at the family level to unassigned
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy[which(is.na(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy$Family)),1]<-"Unassigned"

# Changing row names to the family assignment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy)<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy$Family
                                                                                
# Transposing the dataframe and taking family relative abundances by treatment
t.metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund<-as.data.frame(decostand(t(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy,f=is.numeric)),method='total'))

# Transposing the data frame again so that treatments are column names 
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund<-as.data.frame(t(t.metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund))

# Adding a column that is just the family names  
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund$family<-row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund)

# Subsetting the table so that only those ASVs with a total relative abundance greater than 0.1 are included
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund[,1:6])>0.1,]

# Subsetting to create an other column that includes ASVs with a total relative abundance across treatment types less than 0.1
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_other<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund[,1:6])<0.1,]

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_myco<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_other[grep("Mycosphaerellaceae",metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_other$family),]

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_other<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_other[grep("Mycosphaerellaceae",metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_other$family,invert=TRUE),]

# Adding a row to the data frame that is for the other/unassigned category
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten[nrow(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten)+1,1:6]<-colSums(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_other[,1:6])

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten<-rbind(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_myco,metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten)

# Changing the row names and family values for the other group
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten)[13]<-"Other"
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten[13,7]<-"Other"

# Removing f__ label to clean names
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten$family<-gsub("f__","",metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten$family)

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten$family<-factor(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten$family,levels = metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten$family)

# Converting table to long
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten_long<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten%>%
pivot_longer(1:6) %>%
  group_by(family)

is.factor(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten_long$family)

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_long<-gather(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten,key="Treatment",value="Relative.Abundance",1:6)  


# Plotting relative abundance charts in ggplot
ggplot(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten_long,aes(x=name,y=value))+
  geom_col(aes(x=name,y=value,fill=family),linewidth=1,inherit.aes = FALSE,color="black")+
 theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),legend.position = "bottom")+
  scale_x_discrete(labels=c("No Plant + Mn5","No Plant + Control","No Plant + SARM07","Plant + Mn5","Plant + Control","Plant + SARM07"))+
  guides(fill=guide_legend(title = "ASV ID"))+
  scale_fill_manual(values=RColorBrewer::brewer.pal(n=length(unique(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_group_taxonomy_relabund_ten_long$family)),"Set3")[c(1:8,10:13,9)])


# Beta-Diversity for all nonautoclaved samples regardless of sampling date
pcoa_its_within_nonautoclave_d10<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,method="bray",color="septoria",shape="septoria",polygon=TRUE,color_two="plant")
pcoa_its_within_nonautoclave_d10[[1]]
pcoa_its_within_nonautoclave_d10_plot<-pcoa_its_within_nonautoclave_d10[[2]]+
  guides(fill=guide_legend("Plant Presence"),color=guide_legend("Sphaerulina Isolate x Plant Presence"),shape=guide_legend("Sphaerulina Isolate"))+
  scale_fill_discrete(labels=c("No Plant","Plant"))+
  scale_shape_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence) x No Plant","Control x No Plant","SARM07 (Low Virulence) x No Plant","Mn5 (High Virulence) x Plant","Control x Plant","SARM07 (Low Virulence) x Plant"))
pcoa_its_within_nonautoclave_d10_plot

# PERMANOVA
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10$plant*metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10$septoria)

# General Alpha-Diversity
alpha.diversity_nrmop_nonautoclave_d10<-data.frame(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,is.character),q0=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,is.numeric),q=0),q1=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,is.numeric),q=1),q2=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,is.numeric),q=2))

# ANOVA q0
anova.alpha.diversity_nonautoclave_d10_q0<-aov(q0~plant+septoria,data = alpha.diversity_nrmop_nonautoclave_d10)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_nonautoclave_d10_q0)
car::Anova(anova.alpha.diversity_nonautoclave_d10_q0,type="III")

kruskal.test(q0~septoria,data = alpha.diversity_nrmop_nonautoclave_d10)

# ANOVA q1
anova.alpha.diversity_nonautoclave_d10_q1<-aov(q1~plant+septoria,data = alpha.diversity_nrmop_nonautoclave_d10)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_nonautoclave_d10_q1)
car::Anova(anova.alpha.diversity_nonautoclave_d10_q1,type="III")

kruskal.test(q1~septoria,data = alpha.diversity_nrmop_nonautoclave_d10)

# ANOVA q2
anova.alpha.diversity_nonautoclave_d10_q2<-aov(q2~plant+septoria,data = alpha.diversity_nrmop_nonautoclave_d10)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_nonautoclave_d10_q2)
car::Anova(anova.alpha.diversity_nonautoclave_d10_q2,type="III")

kruskal.test(q2~septoria,data = alpha.diversity_nrmop_nonautoclave_d10)

# Converting to long format to plot in ggplot
alpha.diversity_nrmop_nonautoclave_d10_long<-gather(alpha.diversity_nrmop_nonautoclave_d10,key="q",value="index",12:14)

# Plotting in ggplot
alpha.diversity_its_nrmop_nonautoclave_d10_plot<-ggplot(alpha.diversity_nrmop_nonautoclave_d10_long,aes(x=plant,y=index,fill=septoria))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(inherit.aes=FALSE,aes(x=plant,y=index,fill=septoria),shape=21,color="black",alpha=0.9,position=position_jitterdodge(dodge.width = .90))+
  facet_wrap(.~q,scales = "free_y")+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"))+
  scale_x_discrete(labels=c("No Plant","Plant"))+
  ylab("Index Value")+
  xlab("Plant Presence")+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  #geom_segment(data=sig_code_nonautoclave_d10,aes(x=xstart,xend=xend,y=ystart,yend=yend),inherit.aes=FALSE)+
  guides(fill=guide_legend("Septoria Isolate"),color=guide_legend("Septoria Isolate"))
  
  
alpha.diversity_its_nrmop_nonautoclave_d10_plot
```

# PIME Analysis for Day 10 Autoclaved Soils

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave.txt",header=TRUE,sep='\t')

# Removing periods from ASV labels
colnames(rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave)<-sub("\\."," ",colnames(rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave[grep("D10",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave$date),]

# Adding row names to metadata file to allow for creation of phyloseq object
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10)<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10$Row.names

# Creating an OTU table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_otu_table<-phyloseq::otu_table(object=select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,is.numeric),taxa_are_rows = FALSE)

# Creating a taxonomy table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_tax_table<-phyloseq::tax_table(as.matrix(subset(tax.tab_its,row.names(tax.tab_its)%in%colnames(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10,is.numeric)))))

# Creating a metadata file for the phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_metadata<-phyloseq::sample_data(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10[,2:11])

# Creating phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_phyloseq_obj<-phyloseq::phyloseq(rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_otu_table,rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_tax_table,rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_metadata)

# Estimating the out of bag error rate prior to prevalence filtering
pime.oob.error(rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_phyloseq_obj,"septoria")

# Splitting data by sphaerulina and plant treatment
pime.variable.split_nrmop_nonautoclave_d10<-pime.split.by.variable(rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_phyloseq_obj,"septoria")

# Creating prevalence intervals for each sphaerulina/plant treatment combination
pime.prevalence_nrmop_nonautoclave_d10<-pime.prevalence(pime.variable.split_nrmop_nonautoclave_d10)

# Calculating the random forest error rate for each possible prevalence (in increments of 5)
set.seed(42)
pime.best.prevalence_nrmop_nonautoclave_d10<-pime.best.prevalence(pime.prevalence_nrmop_nonautoclave_d10,"septoria")

# Pulling out the important ASVs for the model with a low error rate
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10<-as.data.frame(pime.best.prevalence_nrmop_nonautoclave_d10$Importance$`Prevalence 85`)

# Pulling out the phyloseq object for the model with a low error rate
pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d10<-pime.prevalence_nrmop_nonautoclave_d10$'85'

# Assessing model reliability by assigning treatments randomly to samples and computing error rate (expect that it should be an error rate greater than the model error rate when using the actual sample designations that does not change regardless of prevalence filtering). 
pime.randomized.model_nrmop_nonautoclave_d10<-pime.error.prediction(rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_phyloseq_obj, "sept_plant", bootstrap = 100, parallel = TRUE, max.prev = 95)
pime.randomized.model_nrmop_nonautoclave_d10$Plot

# Assessing model error rates over multiple bootstrap replicates
replicated.oob.error_nrmop_nonautoclave_d10 <- pime.oob.replicate(pime.prevalence_nrmop_nonautoclave_d10, "sept_plant", bootstrap = 100, parallel = TRUE)
replicated.oob.error_nrmop_nonautoclave_d10$Plot

# Pulling out the ASV table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.asv.table_nrmop_nonautoclave_d10<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d10@otu_table)

# Pulling out the sample metadata from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.metadata_nrmop_nonautoclave_d10<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d10@sam_data)

# Pulling out the taxonomy table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.taxonomy_nrmop_nonautoclave_d10<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d10@tax_table)

# Removing the period from the ASV labels to match with the original sequence IDs
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$SequenceID<-sub("\\."," ",pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$SequenceID)

# Creating new labels to use for ASV importance plot that include the sequence id (ASV and number) and the family assignment
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$tax_id<-sub("f__","",paste0(pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$SequenceID,"_",pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$Family))

# Adjusting labels for the plot to make it look slightly more visually appealing.
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$tax_id<-sub("asv_its ","asv.its.",pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$tax_id)

# Creating new Sphaerulina object to match with the object created above
sphaerulina_asvs_named<-c(c("asv.its.360_Mycosphaerellaceae","asv.its.454_Mycosphaerellaceae","asv.its.1237_Mycosphaerellaceae","asv.its.1238_Mycosphaerellaceae","asv.its.1239_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae","asv.its.1249_Mycosphaerellaceae","asv.its.1288_Mycosphaerellaceae","asv.its.1352_Mycosphaerellaceae","asv.its.2539_Mycosphaerellaceae","asv.its.3962_Mycosphaerellaceae"))

# Plotting the ASV contributions to model error rates. This will highlight Sphaerulina ASVs in red
ggplot(data=pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10,aes(x=reorder(tax_id,MeanDecreaseAccuracy),y=MeanDecreaseAccuracy*100))+
  geom_col()+
  ylab("Mean Decrease in Model Accuracy (%)")+
  xlab("ASV")+
  coord_flip()+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),axis.text.y = element_text(color=rev(c(ifelse(pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d10$tax_id%in%sphaerulina_asvs_named,"red","black")))))

pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10<-merge(pime.prevalence.filtered.metadata_nrmop_nonautoclave_d10,pime.prevalence.filtered.asv.table_nrmop_nonautoclave_d10,by.x=0,by.y=0)

# Removing ASVs identified as Sphaerulina by blast to assess effect on PCoA
pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10_no_sphaerulina<-pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10[,!(colnames(pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10)%in%sphaerulina_asvs)]

# PCoA for nonautoclaved soils without sphaerulina
pcoa_its_within_nonautoclave_d10_core<-pcoa_imager(pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10_no_sphaerulina,method = "bray",color="septoria",shape="plant",polygon=TRUE,color_two="septoria")
pcoa_its_within_nonautoclave_d10_core[[1]]

pcoa_its_within_nonautoclave_d10_core_plot<-pcoa_its_within_nonautoclave_d10_core[[2]]+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"),shape=guide_legend("Plant Presence"))+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_shape_discrete(labels=c("No Plant","Plant"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_within_nonautoclave_d10_core_plot

# PERMANOVA for nonautoclaved soils without Sphaerulina
adonis2(select_if(pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10_no_sphaerulina,is.numeric)~pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10_no_sphaerulina$septoria*pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d10_no_sphaerulina$plant)


# PCoA for nonautoclaved soils without sphaerulina
pcoa_its_within_nonautoclave_d10_no_sphaerulina<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_no_sphaerulina,method = "bray",color="septoria",shape="plant",polygon=TRUE,color_two="septoria")
pcoa_its_within_nonautoclave_d10_no_sphaerulina[[1]]

pcoa_its_within_nonautoclave_d10_no_sphaerulina_plot<-pcoa_its_within_nonautoclave_d10_no_sphaerulina[[2]]+
    guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"),shape=guide_legend("Plant Presence"))+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_shape_discrete(labels=c("No Plant","Plant"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_within_nonautoclave_d10_no_sphaerulina_plot

# PERMANOVA for nonautoclaved soils without Sphaerulina
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_no_sphaerulina,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_no_sphaerulina$septoria+metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d10_no_sphaerulina$plant)
```


# Separating the nonautoclaved soil fungal communities by date starting with day 25. 

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave.txt",header=TRUE,sep='\t')

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,by.x=0,by.y=0)

# Pulling out day 10 samples only
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave[grep("D25",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave$date),]

# Tabulating summaries
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25 %>%
  group_by(date,septoria,plant,autoclave) %>%
  dplyr::count()%>%
  print(n=55)

# Constructing relative abundance stacked bar charts at the family level. First taking the sum of all ASVs by sphaerulina x plant treatment
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25[,11:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25)]%>%
  as.data.frame()%>%
  group_by(sept_plant)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing row names to treatment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group)<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group$sept_plant
colnames(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group)<-sub("\\."," ",colnames(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group))

# Merging in the taxonomy information
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy<-merge(tax.tab_its,as.data.frame(t(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group)),by=0)

# Converting count data to numeric 
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy)]<-lapply(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy[,9:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy)],as.numeric)

# Taking the sum of each family
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy[,c(6,9:length(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy))]%>%
  as.data.frame()%>%
  group_by(Family)%>%
  summarize_all(sum)%>%
  as.data.frame()

# Changing any NAs at the family level to unassigned
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy[which(is.na(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy$Family)),1]<-"Unassigned"

# Changing row names to the family assignment
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy)<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy$Family
                                                                                
# Transposing the dataframe and taking family relative abundances by treatment
t.metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund<-as.data.frame(decostand(t(Filter(x=metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy,f=is.numeric)),method='total'))

# Transposing the data frame again so that treatments are column names 
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund<-as.data.frame(t(t.metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund))

# Adding a column that is just the family names  
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund$family<-row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund)

# Subsetting the table so that only those ASVs with a total relative abundance greater than 0.1 are included
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund[,1:6])>0.1,]

# Subsetting to create an other column that includes ASVs with a total relative abundance across treatment types less than 0.1
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_other<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund[rowSums(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund[,1:6])<0.1,]

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_myco<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_other[grep("Mycosphaerellaceae",metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_other$family),]

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_other<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_other[grep("Mycosphaerellaceae",metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_other$family,invert=TRUE),]

# Adding a row to the data frame that is for the other/unassigned category
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten[nrow(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten)+1,1:6]<-colSums(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_other[,1:6])

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten<-rbind(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_myco,metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten)

# Changing the row names and family values for the other group
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten)[13]<-"Other"
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten[13,7]<-"Other"

# Removing f__ label to clean names
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten$family<-gsub("f__","",metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten$family)

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten$family<-factor(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten$family,levels = metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten$family)

# Converting table to long
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten_long<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten%>%
pivot_longer(1:6) %>%
  group_by(family)

is.factor(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten_long$family)

metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_long<-gather(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten,key="Treatment",value="Relative.Abundance",1:6)  


# Plotting relative abundance charts in ggplot
ggplot(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten_long,aes(x=name,y=value))+
  geom_col(aes(x=name,y=value,fill=family),linewidth=1,inherit.aes = FALSE,color="black")+
 theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),legend.position = "bottom")+
  scale_x_discrete(labels=c("No Plant + Mn5","No Plant + Control","No Plant + SARM07","Plant + Mn5","Plant + Control","Plant + SARM07"))+
  guides(fill=guide_legend(title = "ASV ID"))+
  scale_fill_manual(values=RColorBrewer::brewer.pal(n=length(unique(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_group_taxonomy_relabund_ten_long$family)),"Set3")[c(1:8,10:13,9)])


# Beta-Diversity for all nonautoclaved samples regardless of sampling date
pcoa_its_within_nonautoclave_d25<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,method="bray",color="septoria",shape="plant",polygon=TRUE,color_two="septoria")
pcoa_its_within_nonautoclave_d25[[1]]
pcoa_its_within_nonautoclave_d25_plot<-pcoa_its_within_nonautoclave_d25[[2]]+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"),shape=guide_legend("Plant Presence"))+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_shape_discrete(labels=c("No Plant","Plant"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_within_nonautoclave_d25_plot

# PERMANOVA
adonis2(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,is.numeric)~metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25$plant*metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25$septoria)

# General Alpha-Diversity
alpha.diversity_nrmop_nonautoclave_d25<-data.frame(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,is.character),q0=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,is.numeric),q=0),q1=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,is.numeric),q=1),q2=hill_taxa(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,is.numeric),q=2))

# ANOVA q0
anova.alpha.diversity_nonautoclave_d25_q0<-aov(q0~plant+septoria,data = alpha.diversity_nrmop_nonautoclave_d25)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_nonautoclave_d25_q0)
car::Anova(anova.alpha.diversity_nonautoclave_d25_q0,type="III")

kruskal.test(q0~septoria,data = alpha.diversity_nrmop_nonautoclave_d25)
dunnTest(q0~septoria,data = alpha.diversity_nrmop_nonautoclave_d25)

# ANOVA q1
anova.alpha.diversity_nonautoclave_d25_q1<-aov(q1~plant+septoria,data = alpha.diversity_nrmop_nonautoclave_d25)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_nonautoclave_d25_q1)
car::Anova(anova.alpha.diversity_nonautoclave_d25_q1,type="III")
TukeyHSD(anova.alpha.diversity_nonautoclave_d25_q1)

kruskal.test(q1~septoria,data = alpha.diversity_nrmop_nonautoclave_d25)
dunnTest(q1~septoria,data = alpha.diversity_nrmop_nonautoclave_d25)

# ANOVA q2
anova.alpha.diversity_nonautoclave_d25_q2<-aov(log(q2)~plant+septoria,data = alpha.diversity_nrmop_nonautoclave_d25)
par(mfrow=c(2,2))
plot(anova.alpha.diversity_nonautoclave_d25_q2)
car::Anova(anova.alpha.diversity_nonautoclave_d25_q2,type="III")
TukeyHSD(anova.alpha.diversity_nonautoclave_d25_q2)

kruskal.test(q2~septoria,data = alpha.diversity_nrmop_nonautoclave_d25)
dunnTest(q2~septoria,data = alpha.diversity_nrmop_nonautoclave_d25)

# Converting to long format to plot in ggplot
alpha.diversity_nrmop_nonautoclave_d25_long<-gather(alpha.diversity_nrmop_nonautoclave_d25,key="q",value="index",12:14)

# Creating an object that contains the significance coding based on ANOVAs
sig_code_nonautoclave_d25<-data.frame(q=c("q0","q0","q0","q1","q1","q1","q2","q2","q2"),lab=c(NA,NA,NA,"A","AB","B","A","AB","B"),x=c(1,2,3,1,2,3,1,2,3),y=c(850,850,850,100,100,100,35,35,35))

# Plotting in ggplot
alpha.diversity_its_nrmop_nonautoclave_d25_plot<-ggplot(alpha.diversity_nrmop_nonautoclave_d25_long,aes(x=septoria,y=index,fill=septoria))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(inherit.aes=FALSE,aes(x=septoria,y=index,fill=septoria,shape=plant),size=2.5,color="black",alpha=0.9,position=position_jitterdodge(dodge.width = .90))+
  scale_shape_manual(values=c(21,22),labels=c("No Plant","Plant"))+
  facet_wrap(.~q,scales = "free_y")+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"))+
  geom_text(data=sig_code_nonautoclave_d25,aes(x=x,y=y,label=lab),inherit.aes=FALSE)+
  scale_x_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  ylab("Index Value")+
  xlab("Sphaerulina Isolate")+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate",shape=guide_legend("Plant Presence")))
  
  
alpha.diversity_its_nrmop_nonautoclave_d25_plot
```

# PIME Analysis for Day 25 Autoclaved Soils

```{r}
# loading in rarefied autoclaved ASV table
rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-read.table("~/OneDrive - Oak Ridge National Laboratory/septoria_in_a_bag/its_data/rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave.txt",header=TRUE,sep='\t')

# Removing periods from ASV labels
colnames(rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave)<-sub("\\."," ",colnames(rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave))

# Adding metadata
metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave<-merge(sample.metadata,rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave,by.x=0,by.y=0)

# Pulling out day 25 samples only
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25<-metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave[grep("D25",metadata.rare.asv.tab.fmt_its_d10d25_nrmop_nonautoclave$date),]

# Adding row names to metadata file to allow for creation of phyloseq object
row.names(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25)<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25$Row.names

# Creating an OTU table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_otu_table<-phyloseq::otu_table(object=select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,is.numeric),taxa_are_rows = FALSE)

# Creating a taxonomy table for the phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_tax_table<-phyloseq::tax_table(as.matrix(subset(tax.tab_its,row.names(tax.tab_its)%in%colnames(select_if(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25,is.numeric)))))

# Creating a metadata file for the phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_metadata<-phyloseq::sample_data(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25[,2:11])

# Creating phyloseq object
rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_phyloseq_obj<-phyloseq::phyloseq(rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_otu_table,rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_tax_table,rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_metadata)

# Estimating the out of bag error rate prior to prevalence filtering
pime.oob.error(rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_phyloseq_obj,"septoria")

# Splitting data by sphaerulina and plant treatment
pime.variable.split_nrmop_nonautoclave_d25<-pime.split.by.variable(rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_phyloseq_obj,"septoria")

# Creating prevalence intervals for each sphaerulina/plant treatment combination
pime.prevalence_nrmop_nonautoclave_d25<-pime.prevalence(pime.variable.split_nrmop_nonautoclave_d25)

# Calculating the random forest error rate for each possible prevalence (in increments of 5)
set.seed(42)
pime.best.prevalence_nrmop_nonautoclave_d25<-pime.best.prevalence(pime.prevalence_nrmop_nonautoclave_d25,"septoria")

# Pulling out the important ASVs for the model with a low error rate
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25<-as.data.frame(pime.best.prevalence_nrmop_nonautoclave_d25$Importance$`Prevalence 85`)

# Pulling out the phyloseq object for the model with a low error rate
pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d25<-pime.prevalence_nrmop_nonautoclave_d25$'85'

# Assessing model reliability by assigning treatments randomly to samples and computing error rate (expect that it should be an error rate greater than the model error rate when using the actual sample designations that does not change regardless of prevalence filtering). 
pime.randomized.model_nrmop_nonautoclave_d25<-pime.error.prediction(rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_phyloseq_obj, "sept_plant", bootstrap = 100, parallel = TRUE, max.prev = 95)
pime.randomized.model_nrmop_nonautoclave_d25$Plot

# Assessing model error rates over multiple bootstrap replicates
replicated.oob.error_nrmop_nonautoclave_d25 <- pime.oob.replicate(pime.prevalence_nrmop_nonautoclave_d25, "sept_plant", bootstrap = 100, parallel = TRUE)
replicated.oob.error_nrmop_nonautoclave_d25$Plot

# Pulling out the ASV table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.asv.table_nrmop_nonautoclave_d25<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d25@otu_table)

# Pulling out the sample metadata from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.metadata_nrmop_nonautoclave_d25<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d25@sam_data)

# Pulling out the taxonomy table from the phyloseq object from the prevalence filtered object
pime.prevalence.filtered.taxonomy_nrmop_nonautoclave_d25<-as.data.frame(pime.prevalence.filtered.asv.phyloseq_nrmop_nonautoclave_d25@tax_table)

# Removing the period from the ASV labels to match with the original sequence IDs
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$SequenceID<-sub("\\."," ",pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$SequenceID)

# Creating new labels to use for ASV importance plot that include the sequence id (ASV and number) and the family assignment
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$tax_id<-sub("f__","",paste0(pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$SequenceID,"_",pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$Family))

# Adjusting labels for the plot to make it look slightly more visually appealing.
pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$tax_id<-sub("asv_its ","asv.its.",pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$tax_id)

# Creating new Sphaerulina object to match with the object created above
sphaerulina_asvs_named<-c(c("asv.its.360_Mycosphaerellaceae","asv.its.454_Mycosphaerellaceae","asv.its.1237_Mycosphaerellaceae","asv.its.1238_Mycosphaerellaceae","asv.its.1239_Mycosphaerellaceae","asv.its.1240_Mycosphaerellaceae","asv.its.1249_Mycosphaerellaceae","asv.its.1288_Mycosphaerellaceae","asv.its.1352_Mycosphaerellaceae","asv.its.2539_Mycosphaerellaceae","asv.its.3962_Mycosphaerellaceae"))

# Plotting the ASV contributions to model error rates. This will highlight Sphaerulina ASVs in red
ggplot(data=pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25,aes(x=reorder(tax_id,MeanDecreaseAccuracy),y=MeanDecreaseAccuracy*100))+
  geom_col()+
  ylab("Mean Decrease in Model Accuracy (%)")+
  xlab("ASV")+
  coord_flip()+
  theme(panel.grid=element_blank(),panel.border = element_rect(color="black",fill=NA),panel.background = element_rect(fill="white"),axis.text.y = element_text(color=rev(c(ifelse(pime.prevalence.filtered.important.asvs_nrmop_nonautoclave_d25$tax_id%in%sphaerulina_asvs_named,"red","black")))))

# Removing ASVs identified as Sphaerulina by blast to assess effect on PCoA
metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_no_sphaerulina<-metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25[,!(colnames(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25)%in%sphaerulina_asvs)]

pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25<-merge(pime.prevalence.filtered.metadata_nrmop_nonautoclave_d25,pime.prevalence.filtered.asv.table_nrmop_nonautoclave_d25,by.x=0,by.y=0)

pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25_no_sphaerulina<-pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25[,!(colnames(pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25)%in%sphaerulina_asvs)]

# PCoA for nonautoclaved soils without sphaerulina
pcoa_its_within_nonautoclave_d25_core<-pcoa_imager(pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25_no_sphaerulina,method = "bray",color="septoria",shape="plant",polygon=TRUE,color_two="septoria")
pcoa_its_within_nonautoclave_d25_core[[1]]

pcoa_its_within_nonautoclave_d25_core_plot<-pcoa_its_within_nonautoclave_d25_core[[2]]+
  guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"),shape=guide_legend("Plant Presence"))+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_shape_discrete(labels=c("No Plant","Plant"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_within_nonautoclave_d25_core_plot


# PCoA for nonautoclaved soils without sphaerulina
pcoa_its_within_nonautoclave_d25_no_sphaerulina<-pcoa_imager(metadata.rare.asv.tab.fmt_its_nrmop_nonautoclave_d25_no_sphaerulina,method = "bray",color="septoria",shape="plant",polygon=TRUE,color_two="septoria")
pcoa_its_within_nonautoclave_d25_no_sphaerulina[[1]]

pcoa_its_within_nonautoclave_d25_no_sphaerulina_plot<-pcoa_its_within_nonautoclave_d25_no_sphaerulina[[2]]+
    guides(fill=guide_legend("Sphaerulina Isolate"),color=guide_legend("Sphaerulina Isolate"),shape=guide_legend("Plant Presence"))+
  scale_fill_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))+
  scale_shape_discrete(labels=c("No Plant","Plant"))+
  scale_color_discrete(labels=c("Mn5 (High Virulence)","Control","SARM07 (Low Virulence)"))
pcoa_its_within_nonautoclave_d25_no_sphaerulina_plot

# PERMANOVA for nonautoclaved soils without Sphaerulina
adonis2(select_if(pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25_no_sphaerulina,is.numeric)~pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25_no_sphaerulina$septoria*pime.prevalence.filtered.metadata.asv.table_nrmop_nonautoclave_d25_no_sphaerulina$plant)
```


